{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "de0bd2f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1704ded0",
   "metadata": {},
   "source": [
    "# P4C1T8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "03918ce3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>ill</th>\n",
       "      <th>file_name</th>\n",
       "      <th>project</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>date_collected</th>\n",
       "      <th>time_point_h</th>\n",
       "      <th>cage_id</th>\n",
       "      <th>mouse_id</th>\n",
       "      <th>gender</th>\n",
       "      <th>...</th>\n",
       "      <th>strains</th>\n",
       "      <th>tube_type</th>\n",
       "      <th>tube_weight_empty</th>\n",
       "      <th>tube_weight_full</th>\n",
       "      <th>stool_weight_mg</th>\n",
       "      <th>cfu_g_stool</th>\n",
       "      <th>dna_conc_ng_ul</th>\n",
       "      <th>cfu_incubation_time</th>\n",
       "      <th>cfu_g_stool_spec</th>\n",
       "      <th>cfu_g_stool_cmr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>s1</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s1-A12</td>\n",
       "      <td>P4C1T8</td>\n",
       "      <td>gavage DNA</td>\n",
       "      <td>241217</td>\n",
       "      <td>0 h</td>\n",
       "      <td>c1</td>\n",
       "      <td>mA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1653.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>s2</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s2-B12</td>\n",
       "      <td>P4C1T8</td>\n",
       "      <td>gavage DNA</td>\n",
       "      <td>241217</td>\n",
       "      <td>0 h</td>\n",
       "      <td>c1</td>\n",
       "      <td>mB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1690.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>s3</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s3-C12</td>\n",
       "      <td>P4C1T8</td>\n",
       "      <td>gavage DNA</td>\n",
       "      <td>241217</td>\n",
       "      <td>0 h</td>\n",
       "      <td>c2</td>\n",
       "      <td>mA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1644.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>s4</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s4-D12</td>\n",
       "      <td>P4C1T8</td>\n",
       "      <td>gavage DNA</td>\n",
       "      <td>241217</td>\n",
       "      <td>0 h</td>\n",
       "      <td>c2</td>\n",
       "      <td>mB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1651.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>s5</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s5-E12</td>\n",
       "      <td>P4C1T8</td>\n",
       "      <td>gavage DNA</td>\n",
       "      <td>241217</td>\n",
       "      <td>0 h</td>\n",
       "      <td>c3</td>\n",
       "      <td>mA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1657.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>s6</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s6-F12</td>\n",
       "      <td>P4C1T8</td>\n",
       "      <td>gavage DNA</td>\n",
       "      <td>241217</td>\n",
       "      <td>0 h</td>\n",
       "      <td>c3</td>\n",
       "      <td>mB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1697.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>s7</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s7-G12</td>\n",
       "      <td>P4C1T8</td>\n",
       "      <td>gavage DNA</td>\n",
       "      <td>241217</td>\n",
       "      <td>0 h</td>\n",
       "      <td>c4</td>\n",
       "      <td>mA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1680.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>s8</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s8-H12</td>\n",
       "      <td>P4C1T8</td>\n",
       "      <td>gavage DNA</td>\n",
       "      <td>241217</td>\n",
       "      <td>0 h</td>\n",
       "      <td>c4</td>\n",
       "      <td>mB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1686.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>s9</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s9-H11</td>\n",
       "      <td>P4C1T8</td>\n",
       "      <td>gavage DNA</td>\n",
       "      <td>241217</td>\n",
       "      <td>0 h</td>\n",
       "      <td>c5</td>\n",
       "      <td>mAB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>E. coli|E.coli|B.uniformis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1691.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>s19</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s19-A1</td>\n",
       "      <td>P4C1T8</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>241217</td>\n",
       "      <td>8 h</td>\n",
       "      <td>c1</td>\n",
       "      <td>mA</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>E. coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1674.1</td>\n",
       "      <td>1752.0</td>\n",
       "      <td>77.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows Ã— 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  sample_id     ill file_name project sample_type  date_collected  \\\n",
       "0        s1  ILL138    s1-A12  P4C1T8  gavage DNA          241217   \n",
       "1        s2  ILL138    s2-B12  P4C1T8  gavage DNA          241217   \n",
       "2        s3  ILL138    s3-C12  P4C1T8  gavage DNA          241217   \n",
       "3        s4  ILL138    s4-D12  P4C1T8  gavage DNA          241217   \n",
       "4        s5  ILL138    s5-E12  P4C1T8  gavage DNA          241217   \n",
       "5        s6  ILL138    s6-F12  P4C1T8  gavage DNA          241217   \n",
       "6        s7  ILL138    s7-G12  P4C1T8  gavage DNA          241217   \n",
       "7        s8  ILL138    s8-H12  P4C1T8  gavage DNA          241217   \n",
       "8        s9  ILL138    s9-H11  P4C1T8  gavage DNA          241217   \n",
       "9       s19  ILL138    s19-A1  P4C1T8   stool DNA          241217   \n",
       "\n",
       "  time_point_h cage_id mouse_id  gender  ...                     strains  \\\n",
       "0          0 h      c1       mA     NaN  ...              E. coli|E.coli   \n",
       "1          0 h      c1       mB     NaN  ...              E. coli|E.coli   \n",
       "2          0 h      c2       mA     NaN  ...              E. coli|E.coli   \n",
       "3          0 h      c2       mB     NaN  ...              E. coli|E.coli   \n",
       "4          0 h      c3       mA     NaN  ...              E. coli|E.coli   \n",
       "5          0 h      c3       mB     NaN  ...              E. coli|E.coli   \n",
       "6          0 h      c4       mA     NaN  ...              E. coli|E.coli   \n",
       "7          0 h      c4       mB     NaN  ...              E. coli|E.coli   \n",
       "8          0 h      c5      mAB     NaN  ...  E. coli|E.coli|B.uniformis   \n",
       "9          8 h      c1       mA  female  ...                     E. coli   \n",
       "\n",
       "      tube_type tube_weight_empty tube_weight_full  stool_weight_mg  \\\n",
       "0  2ml_screwcap            1653.5              NaN              NaN   \n",
       "1  2ml_screwcap            1690.5              NaN              NaN   \n",
       "2  2ml_screwcap            1644.7              NaN              NaN   \n",
       "3  2ml_screwcap            1651.9              NaN              NaN   \n",
       "4  2ml_screwcap            1657.2              NaN              NaN   \n",
       "5  2ml_screwcap            1697.4              NaN              NaN   \n",
       "6  2ml_screwcap            1680.2              NaN              NaN   \n",
       "7  2ml_screwcap            1686.0              NaN              NaN   \n",
       "8  2ml_screwcap            1691.0              NaN              NaN   \n",
       "9  2ml_screwcap            1674.1           1752.0             77.9   \n",
       "\n",
       "   cfu_g_stool  dna_conc_ng_ul  cfu_incubation_time  cfu_g_stool_spec  \\\n",
       "0          NaN             NaN                  NaN               NaN   \n",
       "1          NaN             NaN                  NaN               NaN   \n",
       "2          NaN             NaN                  NaN               NaN   \n",
       "3          NaN             NaN                  NaN               NaN   \n",
       "4          NaN             NaN                  NaN               NaN   \n",
       "5          NaN             NaN                  NaN               NaN   \n",
       "6          NaN             NaN                  NaN               NaN   \n",
       "7          NaN             NaN                  NaN               NaN   \n",
       "8          NaN             NaN                  NaN               NaN   \n",
       "9          NaN            4.38                  NaN               NaN   \n",
       "\n",
       "  cfu_g_stool_cmr  \n",
       "0             NaN  \n",
       "1             NaN  \n",
       "2             NaN  \n",
       "3             NaN  \n",
       "4             NaN  \n",
       "5             NaN  \n",
       "6             NaN  \n",
       "7             NaN  \n",
       "8             NaN  \n",
       "9             NaN  \n",
       "\n",
       "[10 rows x 22 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Sample Name/Pool Name*</th>\n",
       "      <th>Library Name*</th>\n",
       "      <th>i7 Index Name*</th>\n",
       "      <th>i7 Index Sequence*</th>\n",
       "      <th>i5 Index Name*</th>\n",
       "      <th>i5 Index Sequence*</th>\n",
       "      <th>Custom Read 1*</th>\n",
       "      <th>Custom Read 2*</th>\n",
       "      <th>Custom Index 1*</th>\n",
       "      <th>Custom Index 2*</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>project</th>\n",
       "      <th>ill</th>\n",
       "      <th>file_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>24</td>\n",
       "      <td>NTC_H3</td>\n",
       "      <td>ILL138_P4C3T2_10k</td>\n",
       "      <td>N7F15</td>\n",
       "      <td>AACGTTAG</td>\n",
       "      <td>S5R08</td>\n",
       "      <td>AACTCTCT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NTC</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>NTC-H3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>48</td>\n",
       "      <td>WATER_D2</td>\n",
       "      <td>ILL139_P4C1T8</td>\n",
       "      <td>N7F26</td>\n",
       "      <td>CAGCGTTG</td>\n",
       "      <td>S5R60</td>\n",
       "      <td>AACCGCAA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WATER</td>\n",
       "      <td>P4C1T8</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>WATER-D2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>53</td>\n",
       "      <td>WATER_B3</td>\n",
       "      <td>ILL139_P4C1T8</td>\n",
       "      <td>N7F27</td>\n",
       "      <td>GCCATAGT</td>\n",
       "      <td>S5R58</td>\n",
       "      <td>TCCGGCGG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WATER</td>\n",
       "      <td>P4C1T8</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>WATER-B3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>75</td>\n",
       "      <td>WATER_C6</td>\n",
       "      <td>ILL139_P4C1T8</td>\n",
       "      <td>N7F30</td>\n",
       "      <td>ATAGAGAC</td>\n",
       "      <td>S5R59</td>\n",
       "      <td>CAATGCCT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WATER</td>\n",
       "      <td>P4C1T8</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>WATER-C6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>99</td>\n",
       "      <td>WATER_F9</td>\n",
       "      <td>ILL139_P4C1T8</td>\n",
       "      <td>N7F33</td>\n",
       "      <td>TAGTAATT</td>\n",
       "      <td>S5R62</td>\n",
       "      <td>GCATGGCG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WATER</td>\n",
       "      <td>P4C1T8</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>WATER-F9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Unnamed: 0 Sample Name/Pool Name*      Library Name* i7 Index Name*  \\\n",
       "23          24                 NTC_H3  ILL138_P4C3T2_10k          N7F15   \n",
       "47          48               WATER_D2      ILL139_P4C1T8          N7F26   \n",
       "52          53               WATER_B3      ILL139_P4C1T8          N7F27   \n",
       "74          75               WATER_C6      ILL139_P4C1T8          N7F30   \n",
       "98          99               WATER_F9      ILL139_P4C1T8          N7F33   \n",
       "\n",
       "   i7 Index Sequence* i5 Index Name* i5 Index Sequence*  Custom Read 1*  \\\n",
       "23           AACGTTAG          S5R08           AACTCTCT             NaN   \n",
       "47           CAGCGTTG          S5R60           AACCGCAA             NaN   \n",
       "52           GCCATAGT          S5R58           TCCGGCGG             NaN   \n",
       "74           ATAGAGAC          S5R59           CAATGCCT             NaN   \n",
       "98           TAGTAATT          S5R62           GCATGGCG             NaN   \n",
       "\n",
       "    Custom Read 2*  Custom Index 1*  Custom Index 2* sample_id project  \\\n",
       "23             NaN              NaN              NaN       NTC  P4C3T2   \n",
       "47             NaN              NaN              NaN     WATER  P4C1T8   \n",
       "52             NaN              NaN              NaN     WATER  P4C1T8   \n",
       "74             NaN              NaN              NaN     WATER  P4C1T8   \n",
       "98             NaN              NaN              NaN     WATER  P4C1T8   \n",
       "\n",
       "       ill file_name  \n",
       "23  ILL138    NTC-H3  \n",
       "47  ILL138  WATER-D2  \n",
       "52  ILL138  WATER-B3  \n",
       "74  ILL138  WATER-C6  \n",
       "98  ILL138  WATER-F9  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ill = pd.read_excel('/Users/ruprec01/Documents/Faith_lab/Git/population_bottlenecks/input_files/ill_sheets/ILL138_GENEWIZ.xls',\n",
    "                            sheet_name = 'Individual Library Information', header = 1)\n",
    "df_samplesheet = pd.read_csv('/Users/ruprec01/Documents/Faith_lab/Git/population_bottlenecks/input_files/metadata/P4C1T8-sample_sheets - P4C1T8.tsv', sep = '\\t')\n",
    "df_samplesheet = df_samplesheet.loc[df_samplesheet['sample_type'].str.contains('DNA')].copy()\n",
    "\n",
    "df_ill = df_ill.dropna(subset=['Sample Name/Pool Name*'])\n",
    "df_ill['sample_id']= df_ill['Sample Name/Pool Name*'].apply(lambda x: str(x).split('_')[0])\n",
    "df_ill['project'] = df_ill['Library Name*'].apply(lambda x: str(x).split('_')[1])\n",
    "#correct library name in ILL138 run\n",
    "df_ill['ill'] = df_ill['Library Name*'].apply(lambda x:  'ILL138' if str(x).split('_')[0] == 'ILL139' else str(x).split('_')[0])\n",
    "df_ill['file_name'] = df_ill['Sample Name/Pool Name*'].apply(lambda x: str(x).replace('_', '-'))\n",
    "df_merge = pd.merge(df_ill[['sample_id', 'ill', 'file_name', 'project']], df_samplesheet, how = 'right', on = ['sample_id', 'project'])\n",
    "display(df_merge.head(10))\n",
    "\n",
    "df_ill_ntc = df_ill.loc[df_ill['Sample Name/Pool Name*'].str.contains(\"WATER|NTC\")]\n",
    "df_ill_ntc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "67ff08c4",
   "metadata": {},
   "source": [
    "## Create readtables for cages\n",
    "- Read tables are created that all samples are quantifed by the mouse gavage. Meaning gavage mA quantifies ST1 and ST2 in all samples!\n",
    "- all cages got uique different gavage solution for every mouse\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "a63a6c46",
   "metadata": {},
   "outputs": [],
   "source": [
    "basepath = '/Volumes/sd/faith/MTCSB/projects/P4-barcoding_strains/r2_data/population-bottlenecks'\n",
    "exportpath = '/Volumes/sd/faith/MTCSB/projects/P4-barcoding_strains/r2_data/population-bottlenecks/fp_inputs/P4C1T8'\n",
    "strains_c1toc4_mAgavage = ['ST1', 'ST2']\n",
    "strains_c1toc4_mBgavage = [ 'ST5', 'ST6']\n",
    "strains_c5 = ['ST1', 'ST2', 'ST4']\n",
    "#get NTC data\n",
    "df_ntc = pd.DataFrame()\n",
    "for idx, row in df_ill_ntc.iterrows():\n",
    "    df = pd.read_csv(basepath + '/' + row['ill'] + '/readtables/' + row['file_name'] + \"/\" + row['file_name'] + '_total_moleculestable.csv')\n",
    "    df['sample'] = row['Sample Name/Pool Name*']+'_'+row['sample_id']\n",
    "    df_ntc = pd.concat([df_ntc, df])\n",
    "df_ntc_pivot = df_ntc.pivot(index = ['strain', 'umi_seq'], columns = 'sample', values = 'molecules').fillna(0)\n",
    "\n",
    "#get cage data\n",
    "for cage in df_merge['cage_id'].unique():\n",
    "    if cage != 'c5':\n",
    "        df_cage = df_merge[df_merge['cage_id'] == cage].copy()\n",
    "        \n",
    "        #creates a readtable per mouse gavage\n",
    "        df_mA_gavage = df_cage[((df_cage['mouse_id'] == 'mA') & (df_cage['sample_type'] == 'gavage DNA'))]\n",
    "        df_mB_gavage = df_cage[((df_cage['mouse_id'] == 'mB') & (df_cage['sample_type'] == 'gavage DNA'))]\n",
    "        for idx, row in df_mA_gavage.iterrows():\n",
    "            df = pd.read_csv(basepath + '/' + row['ill'] + '/readtables/' + row['file_name'] + \"/\" + row['file_name'] + '_total_moleculestable.csv')\n",
    "            df['sample'] = row['sample_id']+'_'+row['sample_type']\n",
    "            df_reads_mA = df\n",
    "        for idx, row in df_mB_gavage.iterrows():\n",
    "            df = pd.read_csv(basepath + '/' + row['ill'] + '/readtables/' + row['file_name'] + \"/\" + row['file_name'] + '_total_moleculestable.csv')\n",
    "            df['sample'] = row['sample_id']+'_'+row['sample_type']\n",
    "            df_reads_mB = df\n",
    "\n",
    "        #get all samples into dataframe   \n",
    "        df_samples = df_cage[df_cage['sample_type'] != 'gavage DNA'].copy()\n",
    "        #print(df_samples)\n",
    "        df_reads = pd.DataFrame()\n",
    "        for idx, row in df_samples.iterrows():\n",
    "            df = pd.read_csv(basepath + '/' + row['ill'] + '/readtables/' + row['file_name'] + \"/\" + row['file_name'] + '_total_moleculestable.csv')\n",
    "            df['sample'] = row['sample_id']+'_'+row['sample_type']\n",
    "            df_reads = pd.concat([df_reads,df])\n",
    "\n",
    "        #pivot all dataframes\n",
    "        df_reads_pivot = df_reads.pivot(index = ['strain', 'umi_seq'], columns = 'sample', values = 'molecules').fillna(0)\n",
    "        df_gavage_mA = df_reads_mA.pivot(index = ['strain', 'umi_seq'], columns = 'sample', values = 'molecules').fillna(0)\n",
    "        df_gavage_mB = df_reads_mB.pivot(index = ['strain', 'umi_seq'], columns = 'sample', values = 'molecules').fillna(0)\n",
    "\n",
    "        df_reads_mA = pd.concat([df_gavage_mA, df_reads_pivot, df_ntc_pivot], axis = 1).fillna(0)\n",
    "        df_reads_mB = pd.concat([df_gavage_mB, df_reads_pivot, df_ntc_pivot], axis = 1).fillna(0)\n",
    "        #replace spaces with '-'\n",
    "        df_reads_mA.columns = df_reads_mA.columns.str.replace(' ', '-')\n",
    "        df_reads_mB.columns = df_reads_mB.columns.str.replace(' ', '-')\n",
    "        df_reads_mA.reset_index(inplace = True)\n",
    "        df_reads_mB.reset_index(inplace = True)\n",
    "        \n",
    "        #export csv files\n",
    "        for strain in strains_c1toc4_mAgavage:\n",
    "            df_reads_mA.loc[df_reads_mA['strain'] == strain].drop('strain', axis = 1).to_csv(exportpath + '/' + cage + '_' + strain + '_moleculestable.csv', index = False)\n",
    "        for strain in strains_c1toc4_mBgavage:\n",
    "            df_reads_mB.loc[df_reads_mB['strain'] == strain].drop('strain', axis = 1).to_csv(exportpath + '/' + cage + '_' + strain + '_moleculestable.csv', index = False)        \n",
    "    else:\n",
    "        df_cage = df_merge[df_merge['cage_id'] == cage].copy()\n",
    "        \n",
    "        #creates a readtable per mouse gavage\n",
    "        df_mA_gavage = df_cage[df_cage['sample_type'] == 'gavage DNA']\n",
    "        for idx, row in df_mA_gavage.iterrows():\n",
    "            df = pd.read_csv(basepath + '/' + row['ill'] + '/readtables/' + row['file_name'] + \"/\" + row['file_name'] + '_total_moleculestable.csv')\n",
    "            df['sample'] = row['sample_id']+'_'+row['sample_type']\n",
    "            df_reads_mA = df\n",
    "        \n",
    "        #get all samples into dataframe   \n",
    "        df_samples = df_cage[df_cage['sample_type'] != 'gavage DNA'].copy()\n",
    "        #print(df_samples)\n",
    "        df_reads = pd.DataFrame()\n",
    "        for idx, row in df_samples.iterrows():\n",
    "            df = pd.read_csv(basepath + '/' + row['ill'] + '/readtables/' + row['file_name'] + \"/\" + row['file_name'] + '_total_moleculestable.csv')\n",
    "            df['sample'] = row['sample_id']+'_'+row['sample_type']\n",
    "            df_reads = pd.concat([df_reads,df])\n",
    "\n",
    "        #pivot all dataframes\n",
    "        df_reads_pivot = df_reads.pivot(index = ['strain', 'umi_seq'], columns = 'sample', values = 'molecules').fillna(0)\n",
    "        df_gavage_mA = df_reads_mA.pivot(index = ['strain', 'umi_seq'], columns = 'sample', values = 'molecules').fillna(0)\n",
    "\n",
    "        df_reads_mA = pd.concat([df_gavage_mA, df_reads_pivot, df_ntc_pivot], axis = 1).fillna(0)\n",
    "        #replace spaces with '-'\n",
    "        df_reads_mA.columns = df_reads_mA.columns.str.replace(' ', '-')\n",
    "        df_reads_mA.reset_index(inplace = True)\n",
    "        \n",
    "        #export csv files\n",
    "        for strain in strains_c5:\n",
    "            df_reads_mA.loc[df_reads_mA['strain'] == strain].drop('strain', axis = 1).to_csv(exportpath + '/' + cage + '_' + strain + '_moleculestable.csv', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "fae60351",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "c1_ST1_moleculestable.csv\n",
      "c1_ST2_moleculestable.csv\n",
      "c1_ST5_moleculestable.csv\n",
      "c1_ST6_moleculestable.csv\n",
      "c2_ST1_moleculestable.csv\n",
      "c2_ST2_moleculestable.csv\n",
      "c2_ST5_moleculestable.csv\n",
      "c2_ST6_moleculestable.csv\n",
      "c3_ST1_moleculestable.csv\n",
      "c3_ST2_moleculestable.csv\n",
      "c3_ST5_moleculestable.csv\n",
      "c3_ST6_moleculestable.csv\n",
      "c4_ST1_moleculestable.csv\n",
      "c4_ST2_moleculestable.csv\n",
      "c4_ST5_moleculestable.csv\n",
      "c4_ST6_moleculestable.csv\n",
      "c5_ST1_moleculestable.csv\n",
      "c5_ST2_moleculestable.csv\n",
      "c5_ST4_moleculestable.csv\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "# read all readtables and drop rows in which all gavage columns are 0\n",
    "exportpath = '/Volumes/sd/faith/MTCSB/projects/P4-barcoding_strains/r2_data/population-bottlenecks/fp_inputs/P4C1T8/'\n",
    "\n",
    "for file in os.listdir(exportpath):\n",
    "    if file.endswith('.csv') and not file.startswith('._') and 'onlygavage' not in file:\n",
    "        print(file)\n",
    "        df = pd.read_csv(exportpath + '/' + file)\n",
    "        #get gavage columns\n",
    "        li_gavage = [col for col in df.columns if 'gavage' in col]\n",
    "        #drop rows in which all gavage columns are 0\n",
    "        df = df[df[li_gavage].sum(axis = 1) > 0].copy()\n",
    "        #drop columns that have only 0s\n",
    "        #dropcols = [col for col in df.columns if df[col].sum() == 0]\n",
    "        #if len(dropcols) > 0:\n",
    "        #    print('NO READS IN')\n",
    "        #    print(dropcols)\n",
    "        df.drop(columns=dropcols).to_csv(exportpath + '/' + str(file).split('_moleculestable.csv')[0] + '_onlygavage_moleculestable.csv', index = False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "243a846f",
   "metadata": {},
   "source": [
    "```cmd\n",
    "rsync -avzP \\\n",
    "/Volumes/sd/faith/MTCSB/projects/P4-barcoding_strains/r2_data/population-bottlenecks/fp_inputs/ \\\n",
    "ruprec02@minerva.hpc.mssm.edu:/sc/arion/work/ruprec02/population-bottlenecks/experiments/\n",
    "```\n",
    "rsync -avzP \\\n",
    "ruprec02@minerva.hpc.mssm.edu:/sc/arion/work/ruprec02/population-bottlenecks/experiments/P4C1T8/FP/ \\\n",
    "/Volumes/sd/faith/MTCSB/projects/P4-barcoding_strains/r2_data/population-bottlenecks/FP/P4C1T8\n",
    "\n",
    "\n",
    "rsync -avzP \\\n",
    "ruprec02@minerva.hpc.mssm.edu:/sc/arion/work/ruprec02/population-bottlenecks/ \\\n",
    "/Volumes/sd/faith/MTCSB/projects/P4-barcoding_strains/r2_data/population-bottlenecks/"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5e497155",
   "metadata": {},
   "source": [
    "# P4C2T4 & T5\n",
    "- both experiments share the same gavage solution and should be possible to combine readtables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "2f8f449c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Sample Name/Pool Name*</th>\n",
       "      <th>Library Name*</th>\n",
       "      <th>i7 Index Name*</th>\n",
       "      <th>i7 Index Sequence*</th>\n",
       "      <th>i7 RC</th>\n",
       "      <th>i5 Index Name*</th>\n",
       "      <th>i5 Index Sequence*</th>\n",
       "      <th>i5 RC</th>\n",
       "      <th>Custom Primer Information Information</th>\n",
       "      <th>Custom Read 2*</th>\n",
       "      <th>Custom Index 1*</th>\n",
       "      <th>Custom Index 2*</th>\n",
       "      <th>Unnamed: 13</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>project</th>\n",
       "      <th>ill</th>\n",
       "      <th>file_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>WATER_A2</td>\n",
       "      <td>ILL130_P4C2T4</td>\n",
       "      <td>N7F02</td>\n",
       "      <td>CTCAGGTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S5R01</td>\n",
       "      <td>AGCTTGCG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WATER</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>WATER-A2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>39</td>\n",
       "      <td>NTC_G5</td>\n",
       "      <td>ILL130_P4C2T4</td>\n",
       "      <td>N7F05</td>\n",
       "      <td>ACCAGACC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S5R07</td>\n",
       "      <td>GCGAGTTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NTC</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>NTC-G5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>40</td>\n",
       "      <td>NTC_H5</td>\n",
       "      <td>ILL130_P4C2T4</td>\n",
       "      <td>N7F05</td>\n",
       "      <td>ACCAGACC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S5R08</td>\n",
       "      <td>AACTCTCT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NTC</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>NTC-H5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>49</td>\n",
       "      <td>WATER_A7</td>\n",
       "      <td>ILL130_P4C2T4</td>\n",
       "      <td>N7F07</td>\n",
       "      <td>TTGGCATG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S5R01</td>\n",
       "      <td>AGCTTGCG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WATER</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>WATER-A7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>79</td>\n",
       "      <td>NTC_G10</td>\n",
       "      <td>ILL130_P4C2T4</td>\n",
       "      <td>N7F10</td>\n",
       "      <td>TTATGAAG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S5R07</td>\n",
       "      <td>GCGAGTTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NTC</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>NTC-G10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>80</td>\n",
       "      <td>NTC_H10</td>\n",
       "      <td>ILL130_P4C2T4</td>\n",
       "      <td>N7F10</td>\n",
       "      <td>TTATGAAG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S5R08</td>\n",
       "      <td>AACTCTCT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NTC</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>NTC-H10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Unnamed: 0 Sample Name/Pool Name*  Library Name* i7 Index Name*  \\\n",
       "8            9               WATER_A2  ILL130_P4C2T4          N7F02   \n",
       "38          39                 NTC_G5  ILL130_P4C2T4          N7F05   \n",
       "39          40                 NTC_H5  ILL130_P4C2T4          N7F05   \n",
       "48          49               WATER_A7  ILL130_P4C2T4          N7F07   \n",
       "78          79                NTC_G10  ILL130_P4C2T4          N7F10   \n",
       "79          80                NTC_H10  ILL130_P4C2T4          N7F10   \n",
       "\n",
       "   i7 Index Sequence*  i7 RC i5 Index Name* i5 Index Sequence*  i5 RC  \\\n",
       "8            CTCAGGTC    NaN          S5R01           AGCTTGCG    NaN   \n",
       "38           ACCAGACC    NaN          S5R07           GCGAGTTC    NaN   \n",
       "39           ACCAGACC    NaN          S5R08           AACTCTCT    NaN   \n",
       "48           TTGGCATG    NaN          S5R01           AGCTTGCG    NaN   \n",
       "78           TTATGAAG    NaN          S5R07           GCGAGTTC    NaN   \n",
       "79           TTATGAAG    NaN          S5R08           AACTCTCT    NaN   \n",
       "\n",
       "    Custom Primer Information Information  Custom Read 2*  Custom Index 1*  \\\n",
       "8                                     NaN             NaN              NaN   \n",
       "38                                    NaN             NaN              NaN   \n",
       "39                                    NaN             NaN              NaN   \n",
       "48                                    NaN             NaN              NaN   \n",
       "78                                    NaN             NaN              NaN   \n",
       "79                                    NaN             NaN              NaN   \n",
       "\n",
       "    Custom Index 2*  Unnamed: 13 sample_id project     ill file_name  \n",
       "8               NaN          NaN     WATER  P4C2T4  ILL130  WATER-A2  \n",
       "38              NaN          NaN       NTC  P4C2T4  ILL130    NTC-G5  \n",
       "39              NaN          NaN       NTC  P4C2T4  ILL130    NTC-H5  \n",
       "48              NaN          NaN     WATER  P4C2T4  ILL130  WATER-A7  \n",
       "78              NaN          NaN       NTC  P4C2T4  ILL130   NTC-G10  \n",
       "79              NaN          NaN       NTC  P4C2T4  ILL130   NTC-H10  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Sample Name/Pool Name*</th>\n",
       "      <th>Library Name*</th>\n",
       "      <th>i7 Index Name*</th>\n",
       "      <th>i7 Index Sequence*</th>\n",
       "      <th>i7 RC</th>\n",
       "      <th>i5 Index Name*</th>\n",
       "      <th>i5 Index Sequence*</th>\n",
       "      <th>i5 RC</th>\n",
       "      <th>Custom Primer Information Information</th>\n",
       "      <th>Custom Read 2*</th>\n",
       "      <th>Custom Index 1*</th>\n",
       "      <th>Custom Index 2*</th>\n",
       "      <th>Unnamed: 13</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>project</th>\n",
       "      <th>ill</th>\n",
       "      <th>file_name</th>\n",
       "      <th>Custom Read 1*</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>WATER_A2</td>\n",
       "      <td>ILL130_P4C2T4</td>\n",
       "      <td>N7F02</td>\n",
       "      <td>CTCAGGTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S5R01</td>\n",
       "      <td>AGCTTGCG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WATER</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>WATER-A2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>39</td>\n",
       "      <td>NTC_G5</td>\n",
       "      <td>ILL130_P4C2T4</td>\n",
       "      <td>N7F05</td>\n",
       "      <td>ACCAGACC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S5R07</td>\n",
       "      <td>GCGAGTTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NTC</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>NTC-G5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>40</td>\n",
       "      <td>NTC_H5</td>\n",
       "      <td>ILL130_P4C2T4</td>\n",
       "      <td>N7F05</td>\n",
       "      <td>ACCAGACC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S5R08</td>\n",
       "      <td>AACTCTCT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NTC</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>NTC-H5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>49</td>\n",
       "      <td>WATER_A7</td>\n",
       "      <td>ILL130_P4C2T4</td>\n",
       "      <td>N7F07</td>\n",
       "      <td>TTGGCATG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S5R01</td>\n",
       "      <td>AGCTTGCG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WATER</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>WATER-A7</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>79</td>\n",
       "      <td>NTC_G10</td>\n",
       "      <td>ILL130_P4C2T4</td>\n",
       "      <td>N7F10</td>\n",
       "      <td>TTATGAAG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S5R07</td>\n",
       "      <td>GCGAGTTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NTC</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>NTC-G10</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>80</td>\n",
       "      <td>NTC_H10</td>\n",
       "      <td>ILL130_P4C2T4</td>\n",
       "      <td>N7F10</td>\n",
       "      <td>TTATGAAG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S5R08</td>\n",
       "      <td>AACTCTCT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NTC</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>NTC-H10</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>22</td>\n",
       "      <td>NTC_F3</td>\n",
       "      <td>ILL135_P4C2T5</td>\n",
       "      <td>N7F15</td>\n",
       "      <td>AACGTTAG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S5R06</td>\n",
       "      <td>CGCTCAAT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NTC</td>\n",
       "      <td>P4C2T5</td>\n",
       "      <td>ILL135</td>\n",
       "      <td>NTC-F3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Unnamed: 0 Sample Name/Pool Name*  Library Name* i7 Index Name*  \\\n",
       "8            9               WATER_A2  ILL130_P4C2T4          N7F02   \n",
       "38          39                 NTC_G5  ILL130_P4C2T4          N7F05   \n",
       "39          40                 NTC_H5  ILL130_P4C2T4          N7F05   \n",
       "48          49               WATER_A7  ILL130_P4C2T4          N7F07   \n",
       "78          79                NTC_G10  ILL130_P4C2T4          N7F10   \n",
       "79          80                NTC_H10  ILL130_P4C2T4          N7F10   \n",
       "21          22                 NTC_F3  ILL135_P4C2T5          N7F15   \n",
       "\n",
       "   i7 Index Sequence*  i7 RC i5 Index Name* i5 Index Sequence*  i5 RC  \\\n",
       "8            CTCAGGTC    NaN          S5R01           AGCTTGCG    NaN   \n",
       "38           ACCAGACC    NaN          S5R07           GCGAGTTC    NaN   \n",
       "39           ACCAGACC    NaN          S5R08           AACTCTCT    NaN   \n",
       "48           TTGGCATG    NaN          S5R01           AGCTTGCG    NaN   \n",
       "78           TTATGAAG    NaN          S5R07           GCGAGTTC    NaN   \n",
       "79           TTATGAAG    NaN          S5R08           AACTCTCT    NaN   \n",
       "21           AACGTTAG    NaN          S5R06           CGCTCAAT    NaN   \n",
       "\n",
       "    Custom Primer Information Information  Custom Read 2*  Custom Index 1*  \\\n",
       "8                                     NaN             NaN              NaN   \n",
       "38                                    NaN             NaN              NaN   \n",
       "39                                    NaN             NaN              NaN   \n",
       "48                                    NaN             NaN              NaN   \n",
       "78                                    NaN             NaN              NaN   \n",
       "79                                    NaN             NaN              NaN   \n",
       "21                                    NaN             NaN              NaN   \n",
       "\n",
       "    Custom Index 2*  Unnamed: 13 sample_id project     ill file_name  \\\n",
       "8               NaN          NaN     WATER  P4C2T4  ILL130  WATER-A2   \n",
       "38              NaN          NaN       NTC  P4C2T4  ILL130    NTC-G5   \n",
       "39              NaN          NaN       NTC  P4C2T4  ILL130    NTC-H5   \n",
       "48              NaN          NaN     WATER  P4C2T4  ILL130  WATER-A7   \n",
       "78              NaN          NaN       NTC  P4C2T4  ILL130   NTC-G10   \n",
       "79              NaN          NaN       NTC  P4C2T4  ILL130   NTC-H10   \n",
       "21              NaN          NaN       NTC  P4C2T5  ILL135    NTC-F3   \n",
       "\n",
       "    Custom Read 1*  \n",
       "8              NaN  \n",
       "38             NaN  \n",
       "39             NaN  \n",
       "48             NaN  \n",
       "78             NaN  \n",
       "79             NaN  \n",
       "21             NaN  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>ill</th>\n",
       "      <th>file_name</th>\n",
       "      <th>project</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>date_collected</th>\n",
       "      <th>time_point_h</th>\n",
       "      <th>cage_id</th>\n",
       "      <th>mouse_type</th>\n",
       "      <th>pretreatment</th>\n",
       "      <th>...</th>\n",
       "      <th>strain_bcs</th>\n",
       "      <th>strains</th>\n",
       "      <th>tube_type</th>\n",
       "      <th>tube_weight_empty</th>\n",
       "      <th>tube_weight_full</th>\n",
       "      <th>stool_weight_mg</th>\n",
       "      <th>cfu_g_stool</th>\n",
       "      <th>cfu_incubation_time</th>\n",
       "      <th>dna_conc_ng_ul</th>\n",
       "      <th>dna_conc_ng_ul.1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>s1</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>s1-A1</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>250401.0</td>\n",
       "      <td>5 h</td>\n",
       "      <td>cage1</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1651.8</td>\n",
       "      <td>1693.0</td>\n",
       "      <td>41.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.040486</td>\n",
       "      <td>0.398801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>s1</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>s1-A6</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>250401.0</td>\n",
       "      <td>5 h</td>\n",
       "      <td>cage1</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1651.8</td>\n",
       "      <td>1693.0</td>\n",
       "      <td>41.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.040486</td>\n",
       "      <td>0.398801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>s3</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>s3-B1</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>250401.0</td>\n",
       "      <td>5 h</td>\n",
       "      <td>cage2</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1653.4</td>\n",
       "      <td>1691.6</td>\n",
       "      <td>38.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.013750</td>\n",
       "      <td>0.442757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>s3</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>s3-B6</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>250401.0</td>\n",
       "      <td>5 h</td>\n",
       "      <td>cage2</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1653.4</td>\n",
       "      <td>1691.6</td>\n",
       "      <td>38.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.013750</td>\n",
       "      <td>0.442757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>s5</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>s5-C1</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>250401.0</td>\n",
       "      <td>5 h</td>\n",
       "      <td>cage3</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>donor_BSD2780</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1697.0</td>\n",
       "      <td>1756.5</td>\n",
       "      <td>59.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.632137</td>\n",
       "      <td>0.493706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>s115</td>\n",
       "      <td>ILL135</td>\n",
       "      <td>s115-C8</td>\n",
       "      <td>P4C2T5</td>\n",
       "      <td>Pr. Colon DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage10</td>\n",
       "      <td>SPF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>s117</td>\n",
       "      <td>ILL135</td>\n",
       "      <td>s117-D8</td>\n",
       "      <td>P4C2T5</td>\n",
       "      <td>Pr. Colon DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage11</td>\n",
       "      <td>SPF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>s119</td>\n",
       "      <td>ILL135</td>\n",
       "      <td>s119-E8</td>\n",
       "      <td>P4C2T5</td>\n",
       "      <td>Pr. Colon DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage12</td>\n",
       "      <td>SPF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>s121</td>\n",
       "      <td>ILL135</td>\n",
       "      <td>s121-F8</td>\n",
       "      <td>P4C2T5</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>s122</td>\n",
       "      <td>ILL135</td>\n",
       "      <td>s122-G8</td>\n",
       "      <td>P4C2T5</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>142 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   sample_id     ill file_name project    sample_type  date_collected  \\\n",
       "0         s1  ILL130     s1-A1  P4C2T4      stool DNA        250401.0   \n",
       "1         s1  ILL130     s1-A6  P4C2T4      stool DNA        250401.0   \n",
       "2         s3  ILL130     s3-B1  P4C2T4      stool DNA        250401.0   \n",
       "3         s3  ILL130     s3-B6  P4C2T4      stool DNA        250401.0   \n",
       "4         s5  ILL130     s5-C1  P4C2T4      stool DNA        250401.0   \n",
       "..       ...     ...       ...     ...            ...             ...   \n",
       "58      s115  ILL135   s115-C8  P4C2T5  Pr. Colon DNA             NaN   \n",
       "59      s117  ILL135   s117-D8  P4C2T5  Pr. Colon DNA             NaN   \n",
       "60      s119  ILL135   s119-E8  P4C2T5  Pr. Colon DNA             NaN   \n",
       "61      s121  ILL135   s121-F8  P4C2T5     Gavage DNA             NaN   \n",
       "62      s122  ILL135   s122-G8  P4C2T5     Gavage DNA             NaN   \n",
       "\n",
       "   time_point_h cage_id mouse_type   pretreatment  ...  \\\n",
       "0           5 h   cage1  germ_free            NaN  ...   \n",
       "1           5 h   cage1  germ_free            NaN  ...   \n",
       "2           5 h   cage2  germ_free            NaN  ...   \n",
       "3           5 h   cage2  germ_free            NaN  ...   \n",
       "4           5 h   cage3  germ_free  donor_BSD2780  ...   \n",
       "..          ...     ...        ...            ...  ...   \n",
       "58         24 h  cage10        SPF            NaN  ...   \n",
       "59         24 h  cage11        SPF            NaN  ...   \n",
       "60         24 h  cage12        SPF            NaN  ...   \n",
       "61         24 h     NaN        NaN            NaN  ...   \n",
       "62         24 h     NaN        NaN            NaN  ...   \n",
       "\n",
       "                  strain_bcs  \\\n",
       "0   ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "1   ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "2   ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "3   ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "4   ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "..                       ...   \n",
       "58  ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "59  ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "60  ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "61  ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "62  ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "\n",
       "                                             strains     tube_type  \\\n",
       "0   E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "1   E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "2   E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "3   E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "4   E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "..                                               ...           ...   \n",
       "58  E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "59  E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "60  E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "61  E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "62  E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "\n",
       "   tube_weight_empty tube_weight_full stool_weight_mg  cfu_g_stool  \\\n",
       "0             1651.8           1693.0            41.2          NaN   \n",
       "1             1651.8           1693.0            41.2          NaN   \n",
       "2             1653.4           1691.6            38.2          NaN   \n",
       "3             1653.4           1691.6            38.2          NaN   \n",
       "4             1697.0           1756.5            59.5          NaN   \n",
       "..               ...              ...             ...          ...   \n",
       "58               NaN              NaN             NaN          NaN   \n",
       "59               NaN              NaN             NaN          NaN   \n",
       "60               NaN              NaN             NaN          NaN   \n",
       "61               NaN              NaN             NaN          NaN   \n",
       "62               NaN              NaN             NaN          NaN   \n",
       "\n",
       "    cfu_incubation_time dna_conc_ng_ul  dna_conc_ng_ul.1  \n",
       "0                   NaN      -0.040486          0.398801  \n",
       "1                   NaN      -0.040486          0.398801  \n",
       "2                   NaN      -0.013750          0.442757  \n",
       "3                   NaN      -0.013750          0.442757  \n",
       "4                   NaN       7.632137          0.493706  \n",
       "..                  ...            ...               ...  \n",
       "58                  NaN            NaN               NaN  \n",
       "59                  NaN            NaN               NaN  \n",
       "60                  NaN            NaN               NaN  \n",
       "61                  NaN            NaN               NaN  \n",
       "62                  NaN            NaN               NaN  \n",
       "\n",
       "[142 rows x 23 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['ILL130' nan 'ILL135']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Sample Name/Pool Name*</th>\n",
       "      <th>Library Name*</th>\n",
       "      <th>i7 Index Name*</th>\n",
       "      <th>i7 Index Sequence*</th>\n",
       "      <th>i7 RC</th>\n",
       "      <th>i5 Index Name*</th>\n",
       "      <th>i5 Index Sequence*</th>\n",
       "      <th>i5 RC</th>\n",
       "      <th>Custom Primer Information Information</th>\n",
       "      <th>Custom Read 2*</th>\n",
       "      <th>Custom Index 1*</th>\n",
       "      <th>Custom Index 2*</th>\n",
       "      <th>Unnamed: 13</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>project</th>\n",
       "      <th>ill</th>\n",
       "      <th>file_name</th>\n",
       "      <th>Custom Read 1*</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>WATER_A2</td>\n",
       "      <td>ILL130_P4C2T4</td>\n",
       "      <td>N7F02</td>\n",
       "      <td>CTCAGGTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S5R01</td>\n",
       "      <td>AGCTTGCG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WATER</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>WATER-A2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>39</td>\n",
       "      <td>NTC_G5</td>\n",
       "      <td>ILL130_P4C2T4</td>\n",
       "      <td>N7F05</td>\n",
       "      <td>ACCAGACC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S5R07</td>\n",
       "      <td>GCGAGTTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NTC</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>NTC-G5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>40</td>\n",
       "      <td>NTC_H5</td>\n",
       "      <td>ILL130_P4C2T4</td>\n",
       "      <td>N7F05</td>\n",
       "      <td>ACCAGACC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S5R08</td>\n",
       "      <td>AACTCTCT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NTC</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>NTC-H5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>49</td>\n",
       "      <td>WATER_A7</td>\n",
       "      <td>ILL130_P4C2T4</td>\n",
       "      <td>N7F07</td>\n",
       "      <td>TTGGCATG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S5R01</td>\n",
       "      <td>AGCTTGCG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WATER</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>WATER-A7</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>79</td>\n",
       "      <td>NTC_G10</td>\n",
       "      <td>ILL130_P4C2T4</td>\n",
       "      <td>N7F10</td>\n",
       "      <td>TTATGAAG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S5R07</td>\n",
       "      <td>GCGAGTTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NTC</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>NTC-G10</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>80</td>\n",
       "      <td>NTC_H10</td>\n",
       "      <td>ILL130_P4C2T4</td>\n",
       "      <td>N7F10</td>\n",
       "      <td>TTATGAAG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S5R08</td>\n",
       "      <td>AACTCTCT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NTC</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>NTC-H10</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>22</td>\n",
       "      <td>NTC_F3</td>\n",
       "      <td>ILL135_P4C2T5</td>\n",
       "      <td>N7F15</td>\n",
       "      <td>AACGTTAG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S5R06</td>\n",
       "      <td>CGCTCAAT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NTC</td>\n",
       "      <td>P4C2T5</td>\n",
       "      <td>ILL135</td>\n",
       "      <td>NTC-F3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Unnamed: 0 Sample Name/Pool Name*  Library Name* i7 Index Name*  \\\n",
       "8            9               WATER_A2  ILL130_P4C2T4          N7F02   \n",
       "38          39                 NTC_G5  ILL130_P4C2T4          N7F05   \n",
       "39          40                 NTC_H5  ILL130_P4C2T4          N7F05   \n",
       "48          49               WATER_A7  ILL130_P4C2T4          N7F07   \n",
       "78          79                NTC_G10  ILL130_P4C2T4          N7F10   \n",
       "79          80                NTC_H10  ILL130_P4C2T4          N7F10   \n",
       "21          22                 NTC_F3  ILL135_P4C2T5          N7F15   \n",
       "\n",
       "   i7 Index Sequence*  i7 RC i5 Index Name* i5 Index Sequence*  i5 RC  \\\n",
       "8            CTCAGGTC    NaN          S5R01           AGCTTGCG    NaN   \n",
       "38           ACCAGACC    NaN          S5R07           GCGAGTTC    NaN   \n",
       "39           ACCAGACC    NaN          S5R08           AACTCTCT    NaN   \n",
       "48           TTGGCATG    NaN          S5R01           AGCTTGCG    NaN   \n",
       "78           TTATGAAG    NaN          S5R07           GCGAGTTC    NaN   \n",
       "79           TTATGAAG    NaN          S5R08           AACTCTCT    NaN   \n",
       "21           AACGTTAG    NaN          S5R06           CGCTCAAT    NaN   \n",
       "\n",
       "    Custom Primer Information Information  Custom Read 2*  Custom Index 1*  \\\n",
       "8                                     NaN             NaN              NaN   \n",
       "38                                    NaN             NaN              NaN   \n",
       "39                                    NaN             NaN              NaN   \n",
       "48                                    NaN             NaN              NaN   \n",
       "78                                    NaN             NaN              NaN   \n",
       "79                                    NaN             NaN              NaN   \n",
       "21                                    NaN             NaN              NaN   \n",
       "\n",
       "    Custom Index 2*  Unnamed: 13 sample_id project     ill file_name  \\\n",
       "8               NaN          NaN     WATER  P4C2T4  ILL130  WATER-A2   \n",
       "38              NaN          NaN       NTC  P4C2T4  ILL130    NTC-G5   \n",
       "39              NaN          NaN       NTC  P4C2T4  ILL130    NTC-H5   \n",
       "48              NaN          NaN     WATER  P4C2T4  ILL130  WATER-A7   \n",
       "78              NaN          NaN       NTC  P4C2T4  ILL130   NTC-G10   \n",
       "79              NaN          NaN       NTC  P4C2T4  ILL130   NTC-H10   \n",
       "21              NaN          NaN       NTC  P4C2T5  ILL135    NTC-F3   \n",
       "\n",
       "    Custom Read 1*  \n",
       "8              NaN  \n",
       "38             NaN  \n",
       "39             NaN  \n",
       "48             NaN  \n",
       "78             NaN  \n",
       "79             NaN  \n",
       "21             NaN  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#P4C2T4:\n",
    "df_ill = pd.read_excel('/Users/ruprec01/Documents/Faith_lab/Git/population_bottlenecks/input_files/ill_sheets/ILL130_P4C2T4.xlsx',\n",
    "                            sheet_name = 'Individual Library Information', header = 1)\n",
    "df_samplesheet = pd.read_csv('/Users/ruprec01/Documents/Faith_lab/Git/population_bottlenecks/input_files/metadata/P4C2T4_cr_mouse_sample - P4C2T4.tsv', sep = '\\t')\n",
    "\n",
    "df_samplesheet = df_samplesheet.loc[df_samplesheet['sample_type'].str.contains('DNA')].copy()\n",
    "df_ill = df_ill.dropna(subset=['Sample Name/Pool Name*'])\n",
    "df_ill['sample_id']= df_ill['Sample Name/Pool Name*'].apply(lambda x: str(x).split('_')[0])\n",
    "df_ill['project'] = df_ill['Library Name*'].apply(lambda x: str(x).split('_')[1])\n",
    "#correct library name in ILL138 run\n",
    "df_ill['ill'] = df_ill['Library Name*'].apply(lambda x:  str(x).split('_')[0])\n",
    "df_ill['file_name'] = df_ill['Sample Name/Pool Name*'].apply(lambda x: str(x).replace('_', '-'))\n",
    "df_merge = pd.merge(df_ill[['sample_id', 'ill', 'file_name', 'project']], df_samplesheet, how = 'right', on = ['sample_id', 'project'])\n",
    "df_merge.head(10)\n",
    "df_P4C2T4 = df_merge.copy()\n",
    "df_ill_ntc = df_ill.loc[df_ill['Sample Name/Pool Name*'].str.contains(\"WATER|NTC\")]\n",
    "display(df_ill_ntc)\n",
    "#P4C2T5\n",
    "df_ill = pd.read_excel('/Users/ruprec01/Documents/Faith_lab/Git/population_bottlenecks/input_files/ill_sheets/ILL135_sample_sheet.xlsx',\n",
    "                            sheet_name = 'Individual Library Information', header = 1)\n",
    "df_samplesheet = pd.read_csv('/Users/ruprec01/Documents/Faith_lab/Git/population_bottlenecks/input_files/metadata/P4C2T5_cr_mouse_sample - P4C2T5.tsv', sep = '\\t')\n",
    "\n",
    "df_samplesheet = df_samplesheet.loc[df_samplesheet['sample_type'].str.contains('DNA')].copy()\n",
    "df_ill = df_ill.dropna(subset=['Sample Name/Pool Name*'])\n",
    "df_ill['sample_id']= df_ill['Sample Name/Pool Name*'].apply(lambda x: str(x).split('_')[0])\n",
    "df_ill['project'] = df_ill['Library Name*'].apply(lambda x: str(x).split('_')[1])\n",
    "#correct library name in ILL138 run\n",
    "df_ill['ill'] = df_ill['Library Name*'].apply(lambda x:  str(x).split('_')[0])\n",
    "df_ill['file_name'] = df_ill['Sample Name/Pool Name*'].apply(lambda x: str(x).replace('_', '-'))\n",
    "df_merge = pd.merge(df_ill[['sample_id', 'ill', 'file_name', 'project']], df_samplesheet, how = 'right', on = ['sample_id', 'project'])\n",
    "df_merge.head(10)\n",
    "df_P4C2T5 = df_merge.copy()\n",
    "df_ill_ntc = pd.concat([df_ill_ntc, df_ill.loc[df_ill['Sample Name/Pool Name*'].str.contains(\"WATER|NTC\")]], axis= 0)\n",
    "display(df_ill_ntc)\n",
    "\n",
    "#merge\n",
    "df_merge = pd.concat([df_P4C2T4, df_P4C2T5], axis = 0)\n",
    "display(df_merge)\n",
    "print(df_merge.ill.unique())\n",
    "\n",
    "df_ill_ntc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "36fcc511",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>ill</th>\n",
       "      <th>file_name</th>\n",
       "      <th>project</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>date_collected</th>\n",
       "      <th>time_point_h</th>\n",
       "      <th>cage_id</th>\n",
       "      <th>mouse_type</th>\n",
       "      <th>pretreatment</th>\n",
       "      <th>...</th>\n",
       "      <th>strain_bcs</th>\n",
       "      <th>strains</th>\n",
       "      <th>tube_type</th>\n",
       "      <th>tube_weight_empty</th>\n",
       "      <th>tube_weight_full</th>\n",
       "      <th>stool_weight_mg</th>\n",
       "      <th>cfu_g_stool</th>\n",
       "      <th>cfu_incubation_time</th>\n",
       "      <th>dna_conc_ng_ul</th>\n",
       "      <th>dna_conc_ng_ul.1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>s81</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>s81-B3</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>250402.0</td>\n",
       "      <td>0 h</td>\n",
       "      <td>Gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>200 ul</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.090138</td>\n",
       "      <td>1.219980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>s81</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>s81-B8</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>250402.0</td>\n",
       "      <td>0 h</td>\n",
       "      <td>Gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>200 ul</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.090138</td>\n",
       "      <td>1.219980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>s82</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>s82-A5</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>250402.0</td>\n",
       "      <td>0 h</td>\n",
       "      <td>Gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>200 ul</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.084682</td>\n",
       "      <td>1.203996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>s82</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>s82-A10</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>250402.0</td>\n",
       "      <td>0 h</td>\n",
       "      <td>Gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>200 ul</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.084682</td>\n",
       "      <td>1.203996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>s121</td>\n",
       "      <td>ILL135</td>\n",
       "      <td>s121-F8</td>\n",
       "      <td>P4C2T5</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>s122</td>\n",
       "      <td>ILL135</td>\n",
       "      <td>s122-G8</td>\n",
       "      <td>P4C2T5</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   sample_id     ill file_name project sample_type  date_collected  \\\n",
       "75       s81  ILL130    s81-B3  P4C2T4  Gavage DNA        250402.0   \n",
       "76       s81  ILL130    s81-B8  P4C2T4  Gavage DNA        250402.0   \n",
       "77       s82  ILL130    s82-A5  P4C2T4  Gavage DNA        250402.0   \n",
       "78       s82  ILL130   s82-A10  P4C2T4  Gavage DNA        250402.0   \n",
       "61      s121  ILL135   s121-F8  P4C2T5  Gavage DNA             NaN   \n",
       "62      s122  ILL135   s122-G8  P4C2T5  Gavage DNA             NaN   \n",
       "\n",
       "   time_point_h cage_id mouse_type pretreatment  ...  \\\n",
       "75          0 h  Gavage        NaN          NaN  ...   \n",
       "76          0 h  Gavage        NaN          NaN  ...   \n",
       "77          0 h  Gavage        NaN          NaN  ...   \n",
       "78          0 h  Gavage        NaN          NaN  ...   \n",
       "61         24 h     NaN        NaN          NaN  ...   \n",
       "62         24 h     NaN        NaN          NaN  ...   \n",
       "\n",
       "                  strain_bcs  \\\n",
       "75  ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "76  ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "77  ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "78  ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "61  ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "62  ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "\n",
       "                                             strains     tube_type  \\\n",
       "75  E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "76  E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "77  E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "78  E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "61  E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "62  E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "\n",
       "   tube_weight_empty tube_weight_full stool_weight_mg  cfu_g_stool  \\\n",
       "75               NaN              NaN          200 ul          NaN   \n",
       "76               NaN              NaN          200 ul          NaN   \n",
       "77               NaN              NaN          200 ul          NaN   \n",
       "78               NaN              NaN          200 ul          NaN   \n",
       "61               NaN              NaN             NaN          NaN   \n",
       "62               NaN              NaN             NaN          NaN   \n",
       "\n",
       "    cfu_incubation_time dna_conc_ng_ul  dna_conc_ng_ul.1  \n",
       "75                  NaN      -0.090138          1.219980  \n",
       "76                  NaN      -0.090138          1.219980  \n",
       "77                  NaN      -0.084682          1.203996  \n",
       "78                  NaN      -0.084682          1.203996  \n",
       "61                  NaN            NaN               NaN  \n",
       "62                  NaN            NaN               NaN  \n",
       "\n",
       "[6 rows x 23 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge[df_merge['sample_type'].str.contains('Gavage')]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "5d67e081",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>ill</th>\n",
       "      <th>file_name</th>\n",
       "      <th>project</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>date_collected</th>\n",
       "      <th>time_point_h</th>\n",
       "      <th>cage_id</th>\n",
       "      <th>mouse_type</th>\n",
       "      <th>pretreatment</th>\n",
       "      <th>...</th>\n",
       "      <th>strain_bcs</th>\n",
       "      <th>strains</th>\n",
       "      <th>tube_type</th>\n",
       "      <th>tube_weight_empty</th>\n",
       "      <th>tube_weight_full</th>\n",
       "      <th>stool_weight_mg</th>\n",
       "      <th>cfu_g_stool</th>\n",
       "      <th>cfu_incubation_time</th>\n",
       "      <th>dna_conc_ng_ul</th>\n",
       "      <th>dna_conc_ng_ul.1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>s1</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>s1-A1</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>250401.0</td>\n",
       "      <td>5 h</td>\n",
       "      <td>cage1</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1651.8</td>\n",
       "      <td>1693.0</td>\n",
       "      <td>41.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.040486</td>\n",
       "      <td>0.398801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>s1</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>s1-A6</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>250401.0</td>\n",
       "      <td>5 h</td>\n",
       "      <td>cage1</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1651.8</td>\n",
       "      <td>1693.0</td>\n",
       "      <td>41.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.040486</td>\n",
       "      <td>0.398801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>s3</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>s3-B1</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>250401.0</td>\n",
       "      <td>5 h</td>\n",
       "      <td>cage2</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1653.4</td>\n",
       "      <td>1691.6</td>\n",
       "      <td>38.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.013750</td>\n",
       "      <td>0.442757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>s3</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>s3-B6</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>250401.0</td>\n",
       "      <td>5 h</td>\n",
       "      <td>cage2</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1653.4</td>\n",
       "      <td>1691.6</td>\n",
       "      <td>38.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.013750</td>\n",
       "      <td>0.442757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>s5</td>\n",
       "      <td>ILL130</td>\n",
       "      <td>s5-C1</td>\n",
       "      <td>P4C2T4</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>250401.0</td>\n",
       "      <td>5 h</td>\n",
       "      <td>cage3</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>donor_BSD2780</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1697.0</td>\n",
       "      <td>1756.5</td>\n",
       "      <td>59.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.632137</td>\n",
       "      <td>0.493706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>s111</td>\n",
       "      <td>ILL135</td>\n",
       "      <td>s111-A8</td>\n",
       "      <td>P4C2T5</td>\n",
       "      <td>Pr. Colon DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage8</td>\n",
       "      <td>SPF</td>\n",
       "      <td>antibiotic_vancomycin</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>s113</td>\n",
       "      <td>ILL135</td>\n",
       "      <td>s113-B8</td>\n",
       "      <td>P4C2T5</td>\n",
       "      <td>Pr. Colon DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage9</td>\n",
       "      <td>SPF</td>\n",
       "      <td>antibiotic_vancomycin</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>s115</td>\n",
       "      <td>ILL135</td>\n",
       "      <td>s115-C8</td>\n",
       "      <td>P4C2T5</td>\n",
       "      <td>Pr. Colon DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage10</td>\n",
       "      <td>SPF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>s117</td>\n",
       "      <td>ILL135</td>\n",
       "      <td>s117-D8</td>\n",
       "      <td>P4C2T5</td>\n",
       "      <td>Pr. Colon DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage11</td>\n",
       "      <td>SPF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>s119</td>\n",
       "      <td>ILL135</td>\n",
       "      <td>s119-E8</td>\n",
       "      <td>P4C2T5</td>\n",
       "      <td>Pr. Colon DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage12</td>\n",
       "      <td>SPF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d|ST4a|ST5a|ST6a</td>\n",
       "      <td>E. coli|E.coli|B.uniformis|B.longum|P.distasonis</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>130 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   sample_id     ill file_name project    sample_type  date_collected  \\\n",
       "0         s1  ILL130     s1-A1  P4C2T4      stool DNA        250401.0   \n",
       "1         s1  ILL130     s1-A6  P4C2T4      stool DNA        250401.0   \n",
       "2         s3  ILL130     s3-B1  P4C2T4      stool DNA        250401.0   \n",
       "3         s3  ILL130     s3-B6  P4C2T4      stool DNA        250401.0   \n",
       "4         s5  ILL130     s5-C1  P4C2T4      stool DNA        250401.0   \n",
       "..       ...     ...       ...     ...            ...             ...   \n",
       "56      s111  ILL135   s111-A8  P4C2T5  Pr. Colon DNA             NaN   \n",
       "57      s113  ILL135   s113-B8  P4C2T5  Pr. Colon DNA             NaN   \n",
       "58      s115  ILL135   s115-C8  P4C2T5  Pr. Colon DNA             NaN   \n",
       "59      s117  ILL135   s117-D8  P4C2T5  Pr. Colon DNA             NaN   \n",
       "60      s119  ILL135   s119-E8  P4C2T5  Pr. Colon DNA             NaN   \n",
       "\n",
       "   time_point_h cage_id mouse_type           pretreatment  ...  \\\n",
       "0           5 h   cage1  germ_free                    NaN  ...   \n",
       "1           5 h   cage1  germ_free                    NaN  ...   \n",
       "2           5 h   cage2  germ_free                    NaN  ...   \n",
       "3           5 h   cage2  germ_free                    NaN  ...   \n",
       "4           5 h   cage3  germ_free          donor_BSD2780  ...   \n",
       "..          ...     ...        ...                    ...  ...   \n",
       "56         24 h   cage8        SPF  antibiotic_vancomycin  ...   \n",
       "57         24 h   cage9        SPF  antibiotic_vancomycin  ...   \n",
       "58         24 h  cage10        SPF                    NaN  ...   \n",
       "59         24 h  cage11        SPF                    NaN  ...   \n",
       "60         24 h  cage12        SPF                    NaN  ...   \n",
       "\n",
       "                  strain_bcs  \\\n",
       "0   ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "1   ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "2   ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "3   ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "4   ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "..                       ...   \n",
       "56  ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "57  ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "58  ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "59  ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "60  ST1a|ST2d|ST4a|ST5a|ST6a   \n",
       "\n",
       "                                             strains     tube_type  \\\n",
       "0   E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "1   E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "2   E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "3   E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "4   E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "..                                               ...           ...   \n",
       "56  E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "57  E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "58  E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "59  E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "60  E. coli|E.coli|B.uniformis|B.longum|P.distasonis  2ml_screwcap   \n",
       "\n",
       "   tube_weight_empty tube_weight_full stool_weight_mg  cfu_g_stool  \\\n",
       "0             1651.8           1693.0            41.2          NaN   \n",
       "1             1651.8           1693.0            41.2          NaN   \n",
       "2             1653.4           1691.6            38.2          NaN   \n",
       "3             1653.4           1691.6            38.2          NaN   \n",
       "4             1697.0           1756.5            59.5          NaN   \n",
       "..               ...              ...             ...          ...   \n",
       "56               NaN              NaN             NaN          NaN   \n",
       "57               NaN              NaN             NaN          NaN   \n",
       "58               NaN              NaN             NaN          NaN   \n",
       "59               NaN              NaN             NaN          NaN   \n",
       "60               NaN              NaN             NaN          NaN   \n",
       "\n",
       "    cfu_incubation_time dna_conc_ng_ul  dna_conc_ng_ul.1  \n",
       "0                   NaN      -0.040486          0.398801  \n",
       "1                   NaN      -0.040486          0.398801  \n",
       "2                   NaN      -0.013750          0.442757  \n",
       "3                   NaN      -0.013750          0.442757  \n",
       "4                   NaN       7.632137          0.493706  \n",
       "..                  ...            ...               ...  \n",
       "56                  NaN            NaN               NaN  \n",
       "57                  NaN            NaN               NaN  \n",
       "58                  NaN            NaN               NaN  \n",
       "59                  NaN            NaN               NaN  \n",
       "60                  NaN            NaN               NaN  \n",
       "\n",
       "[130 rows x 23 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# generate readtables for getFP\n",
    "basepath = '/Volumes/sd/faith/MTCSB/projects/P4-barcoding_strains/r2_data/population-bottlenecks'\n",
    "exportpath = '/Volumes/sd/faith/MTCSB/projects/P4-barcoding_strains/r2_data/population-bottlenecks/fp_inputs/P4C2T4T5'\n",
    "li_strains = ['ST1', 'ST2', 'ST4', 'ST5', 'ST6']\n",
    "\n",
    "df_ntc = pd.DataFrame()\n",
    "for idx, row in df_ill_ntc.iterrows():\n",
    "    df = pd.read_csv(basepath + '/' + row['ill'] + '/readtables/' + row['file_name'] + \"/\" + row['file_name'] + '_total_moleculestable.csv')\n",
    "    df['sample'] = row['Sample Name/Pool Name*']+'_'+row['sample_id']\n",
    "    df_ntc = pd.concat([df_ntc, df])\n",
    "df_ntc_pivot = df_ntc.pivot(index = ['strain', 'umi_seq'], columns = 'sample', values = 'molecules').fillna(0)\n",
    "\n",
    "\n",
    "#get gavage reads dataframe\n",
    "#drop files without sequencing data\n",
    "df_merge.dropna(subset=['file_name'], inplace = True)\n",
    "df_gavage = df_merge[df_merge['sample_type'] == 'Gavage DNA'].copy()\n",
    "df_reads_gavage = pd.DataFrame()\n",
    "for idx, row in df_gavage.iterrows():\n",
    "    df = pd.read_csv(basepath + '/' + row['ill'] + '/readtables/' + row['file_name'] + \"/\" + row['file_name'] + '_total_moleculestable.csv')\n",
    "    df['sample'] = row['project']+'_'+row['file_name']+'_'+row['sample_type']\n",
    "    df_reads_gavage = pd.concat([df_reads_gavage,df])\n",
    "\n",
    "#get sample reads dataframe\n",
    "#get all samples into dataframe   \n",
    "df_samples = df_merge[df_merge['sample_type'] != 'Gavage DNA'].copy()\n",
    "display(df_samples)\n",
    "#print(df_samples)\n",
    "df_sample_reads = pd.DataFrame()\n",
    "for idx, row in df_samples.iterrows():\n",
    "    df = pd.read_csv(basepath + '/' + row['ill'] + '/readtables/' + row['file_name'] + \"/\" + row['file_name'] + '_total_moleculestable.csv')\n",
    "    df['sample'] = row['project']+'_'+row['file_name']+'_'+row['sample_type']\n",
    "    df_sample_reads = pd.concat([df_sample_reads,df])\n",
    "\n",
    "#pivot all dataframes\n",
    "df_sample_pivot = df_sample_reads.pivot(index = ['strain', 'umi_seq'], columns = 'sample', values = 'molecules').fillna(0)\n",
    "df_gavage_pivot = df_reads_gavage.pivot(index = ['strain', 'umi_seq'], columns = 'sample', values = 'molecules').fillna(0)\n",
    "\n",
    "df_readstable = pd.concat([df_gavage_pivot, df_sample_pivot, df_ntc_pivot], axis = 1).fillna(0)\n",
    "\n",
    "df_readstable.columns = df_readstable.columns.str.replace(' ', '-')\n",
    "df_readstable.reset_index(inplace = True)\n",
    "\n",
    "for strain in li_strains:\n",
    "    df_readstable.loc[df_readstable['strain'] == strain].drop('strain', axis = 1).to_csv(exportpath + '/' + strain + '_moleculestable.csv', index = False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "27833836",
   "metadata": {},
   "outputs": [],
   "source": [
    "# read all readtables and drop rows in which all gavage columns are 0\n",
    "li_strains = ['ST1', 'ST2', 'ST4', 'ST5', 'ST6']\n",
    "exportpath = '/Volumes/sd/faith/MTCSB/projects/P4-barcoding_strains/r2_data/population-bottlenecks/fp_inputs/P4C2T4T5'\n",
    "\n",
    "for strain in li_strains:\n",
    "    df = pd.read_csv(exportpath + '/' + strain + '_moleculestable.csv')\n",
    "    #get gavage columns\n",
    "    li_gavage = [col for col in df.columns if 'Gavage' in col]\n",
    "    #drop rows in which all gavage columns are 0\n",
    "    df[df[li_gavage].sum(axis = 1) > 0].to_csv(exportpath + '/' + strain + '_onlygavage_moleculestable.csv', index = False)\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3f21188c",
   "metadata": {},
   "source": [
    "rsync -avzP \\\n",
    "/Volumes/sd/faith/MTCSB/projects/P4-barcoding_strains/r2_data/population-bottlenecks/fp_inputs/P4C1T4T5/ \\\n",
    "ruprec02@minerva.hpc.mssm.edu:/sc/arion/work/ruprec02/population-bottlenecks/experiments/P4C1T4T5/fp_inputs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2a1c4e56",
   "metadata": {},
   "source": [
    "# P4C3T1 100k barcodes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "7ea0ef10",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>ill</th>\n",
       "      <th>file_name</th>\n",
       "      <th>project</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>date_collected</th>\n",
       "      <th>time_point_h</th>\n",
       "      <th>cage_id</th>\n",
       "      <th>mouse_type</th>\n",
       "      <th>pretreatment</th>\n",
       "      <th>...</th>\n",
       "      <th>strain_bcs</th>\n",
       "      <th>strains</th>\n",
       "      <th>tube_type</th>\n",
       "      <th>tube_weight_empty</th>\n",
       "      <th>tube_weight_full</th>\n",
       "      <th>stool_weight_mg</th>\n",
       "      <th>cfu_g_stool</th>\n",
       "      <th>cfu_incubation_time</th>\n",
       "      <th>dna_conc_ng_ul</th>\n",
       "      <th>dna_conc_ng_ul.1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>s1</td>\n",
       "      <td>ILL134</td>\n",
       "      <td>s1-A1</td>\n",
       "      <td>P4C3T1</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5 h</td>\n",
       "      <td>cage1</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1643.5</td>\n",
       "      <td>1681.8</td>\n",
       "      <td>38.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.221408</td>\n",
       "      <td>1.809577</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>s1</td>\n",
       "      <td>ILL134</td>\n",
       "      <td>s1-A6</td>\n",
       "      <td>P4C3T1</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5 h</td>\n",
       "      <td>cage1</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1643.5</td>\n",
       "      <td>1681.8</td>\n",
       "      <td>38.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.221408</td>\n",
       "      <td>1.809577</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>s3</td>\n",
       "      <td>ILL134</td>\n",
       "      <td>s3-B1</td>\n",
       "      <td>P4C3T1</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5 h</td>\n",
       "      <td>cage2</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1659.5</td>\n",
       "      <td>1707.5</td>\n",
       "      <td>48.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.945352</td>\n",
       "      <td>1.420845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>s3</td>\n",
       "      <td>ILL134</td>\n",
       "      <td>s3-B6</td>\n",
       "      <td>P4C3T1</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5 h</td>\n",
       "      <td>cage2</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1659.5</td>\n",
       "      <td>1707.5</td>\n",
       "      <td>48.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.945352</td>\n",
       "      <td>1.420845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>s5</td>\n",
       "      <td>ILL134</td>\n",
       "      <td>s5-C1</td>\n",
       "      <td>P4C3T1</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5 h</td>\n",
       "      <td>cage3</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1706.8</td>\n",
       "      <td>1744.8</td>\n",
       "      <td>38.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.159437</td>\n",
       "      <td>1.792676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>s63</td>\n",
       "      <td>ILL134</td>\n",
       "      <td>s63-E1</td>\n",
       "      <td>P4C3T1</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>s63</td>\n",
       "      <td>ILL134</td>\n",
       "      <td>s63-E6</td>\n",
       "      <td>P4C3T1</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>s64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P4C3T1</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>s65</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P4C3T1</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>s66</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P4C3T1</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>68 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   sample_id     ill file_name project sample_type  date_collected  \\\n",
       "0         s1  ILL134     s1-A1  P4C3T1   stool DNA             NaN   \n",
       "1         s1  ILL134     s1-A6  P4C3T1   stool DNA             NaN   \n",
       "2         s3  ILL134     s3-B1  P4C3T1   stool DNA             NaN   \n",
       "3         s3  ILL134     s3-B6  P4C3T1   stool DNA             NaN   \n",
       "4         s5  ILL134     s5-C1  P4C3T1   stool DNA             NaN   \n",
       "..       ...     ...       ...     ...         ...             ...   \n",
       "63       s63  ILL134    s63-E1  P4C3T1  Gavage DNA             NaN   \n",
       "64       s63  ILL134    s63-E6  P4C3T1  Gavage DNA             NaN   \n",
       "65       s64     NaN       NaN  P4C3T1  Gavage DNA             NaN   \n",
       "66       s65     NaN       NaN  P4C3T1  Gavage DNA             NaN   \n",
       "67       s66     NaN       NaN  P4C3T1  Gavage DNA             NaN   \n",
       "\n",
       "   time_point_h cage_id mouse_type  pretreatment  ... strain_bcs  \\\n",
       "0           5 h   cage1  germ_free           NaN  ...  ST1a|ST2d   \n",
       "1           5 h   cage1  germ_free           NaN  ...  ST1a|ST2d   \n",
       "2           5 h   cage2  germ_free           NaN  ...  ST1a|ST2d   \n",
       "3           5 h   cage2  germ_free           NaN  ...  ST1a|ST2d   \n",
       "4           5 h   cage3  germ_free           NaN  ...  ST1a|ST2d   \n",
       "..          ...     ...        ...           ...  ...        ...   \n",
       "63         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "64         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "65         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "66         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "67         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "\n",
       "           strains     tube_type tube_weight_empty tube_weight_full  \\\n",
       "0   E. coli|E.coli  2ml_screwcap            1643.5           1681.8   \n",
       "1   E. coli|E.coli  2ml_screwcap            1643.5           1681.8   \n",
       "2   E. coli|E.coli  2ml_screwcap            1659.5           1707.5   \n",
       "3   E. coli|E.coli  2ml_screwcap            1659.5           1707.5   \n",
       "4   E. coli|E.coli  2ml_screwcap            1706.8           1744.8   \n",
       "..             ...           ...               ...              ...   \n",
       "63  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "64  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "65  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "66  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "67  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "\n",
       "   stool_weight_mg  cfu_g_stool  cfu_incubation_time  dna_conc_ng_ul  \\\n",
       "0             38.3          NaN                  NaN       -1.221408   \n",
       "1             38.3          NaN                  NaN       -1.221408   \n",
       "2             48.0          NaN                  NaN       -0.945352   \n",
       "3             48.0          NaN                  NaN       -0.945352   \n",
       "4             38.0          NaN                  NaN       -1.159437   \n",
       "..             ...          ...                  ...             ...   \n",
       "63             NaN          NaN                  NaN             NaN   \n",
       "64             NaN          NaN                  NaN             NaN   \n",
       "65             NaN          NaN                  NaN             NaN   \n",
       "66             NaN          NaN                  NaN             NaN   \n",
       "67             NaN          NaN                  NaN             NaN   \n",
       "\n",
       "    dna_conc_ng_ul.1  \n",
       "0           1.809577  \n",
       "1           1.809577  \n",
       "2           1.420845  \n",
       "3           1.420845  \n",
       "4           1.792676  \n",
       "..               ...  \n",
       "63               NaN  \n",
       "64               NaN  \n",
       "65               NaN  \n",
       "66               NaN  \n",
       "67               NaN  \n",
       "\n",
       "[68 rows x 23 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#P4C2T4:\n",
    "df_ill = pd.read_excel('/Users/ruprec01/Documents/Faith_lab/Git/population_bottlenecks/input_files/ill_sheets/ILL134_sample_sheet.xlsx',\n",
    "                            sheet_name = 'Individual Library Information', header = 1)\n",
    "df_samplesheet = pd.read_csv('/Users/ruprec01/Documents/Faith_lab/Git/population_bottlenecks/input_files/metadata/P4C3T1_cr_mouse_sample - P4C3T1.tsv', sep = '\\t')\n",
    "\n",
    "df_samplesheet = df_samplesheet.loc[df_samplesheet['sample_type'].str.contains('DNA')].copy()\n",
    "df_ill = df_ill.dropna(subset=['Sample Name/Pool Name*'])\n",
    "df_ill['sample_id']= df_ill['Sample Name/Pool Name*'].apply(lambda x: str(x).split('_')[0])\n",
    "df_ill['project'] = df_ill['Library Name*'].apply(lambda x: str(x).split('_')[1])\n",
    "#correct library name in ILL138 run\n",
    "df_ill['ill'] = df_ill['Library Name*'].apply(lambda x:  str(x).split('_')[0])\n",
    "df_ill['file_name'] = df_ill['Sample Name/Pool Name*'].apply(lambda x: str(x).replace('_', '-'))\n",
    "df_merge = pd.merge(df_ill[['sample_id', 'ill', 'file_name', 'project']], df_samplesheet, how = 'right', on = ['sample_id', 'project'])\n",
    "df_merge.head(10)\n",
    "\n",
    "df_ill_ntc = df_ill.loc[df_ill['Sample Name/Pool Name*'].str.contains(\"WATER|NTC\")]\n",
    "display(df_merge)\n",
    "df_merge = df_merge.dropna(subset= ['file_name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "bb405a54",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10^8\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>ill</th>\n",
       "      <th>file_name</th>\n",
       "      <th>project</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>date_collected</th>\n",
       "      <th>time_point_h</th>\n",
       "      <th>cage_id</th>\n",
       "      <th>mouse_type</th>\n",
       "      <th>pretreatment</th>\n",
       "      <th>...</th>\n",
       "      <th>strain_bcs</th>\n",
       "      <th>strains</th>\n",
       "      <th>tube_type</th>\n",
       "      <th>tube_weight_empty</th>\n",
       "      <th>tube_weight_full</th>\n",
       "      <th>stool_weight_mg</th>\n",
       "      <th>cfu_g_stool</th>\n",
       "      <th>cfu_incubation_time</th>\n",
       "      <th>dna_conc_ng_ul</th>\n",
       "      <th>dna_conc_ng_ul.1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>s61</td>\n",
       "      <td>ILL134</td>\n",
       "      <td>s61-H4</td>\n",
       "      <td>P4C3T1</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>s61</td>\n",
       "      <td>ILL134</td>\n",
       "      <td>s61-H9</td>\n",
       "      <td>P4C3T1</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>s62</td>\n",
       "      <td>ILL134</td>\n",
       "      <td>s62-A5</td>\n",
       "      <td>P4C3T1</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>s62</td>\n",
       "      <td>ILL134</td>\n",
       "      <td>s62-A10</td>\n",
       "      <td>P4C3T1</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   sample_id     ill file_name project sample_type  date_collected  \\\n",
       "59       s61  ILL134    s61-H4  P4C3T1  Gavage DNA             NaN   \n",
       "60       s61  ILL134    s61-H9  P4C3T1  Gavage DNA             NaN   \n",
       "61       s62  ILL134    s62-A5  P4C3T1  Gavage DNA             NaN   \n",
       "62       s62  ILL134   s62-A10  P4C3T1  Gavage DNA             NaN   \n",
       "\n",
       "   time_point_h cage_id mouse_type  pretreatment  ... strain_bcs  \\\n",
       "59         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "60         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "61         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "62         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "\n",
       "           strains     tube_type tube_weight_empty tube_weight_full  \\\n",
       "59  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "60  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "61  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "62  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "\n",
       "   stool_weight_mg  cfu_g_stool  cfu_incubation_time  dna_conc_ng_ul  \\\n",
       "59             NaN          NaN                  NaN             NaN   \n",
       "60             NaN          NaN                  NaN             NaN   \n",
       "61             NaN          NaN                  NaN             NaN   \n",
       "62             NaN          NaN                  NaN             NaN   \n",
       "\n",
       "    dna_conc_ng_ul.1  \n",
       "59               NaN  \n",
       "60               NaN  \n",
       "61               NaN  \n",
       "62               NaN  \n",
       "\n",
       "[4 rows x 23 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10^6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>ill</th>\n",
       "      <th>file_name</th>\n",
       "      <th>project</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>date_collected</th>\n",
       "      <th>time_point_h</th>\n",
       "      <th>cage_id</th>\n",
       "      <th>mouse_type</th>\n",
       "      <th>pretreatment</th>\n",
       "      <th>...</th>\n",
       "      <th>strain_bcs</th>\n",
       "      <th>strains</th>\n",
       "      <th>tube_type</th>\n",
       "      <th>tube_weight_empty</th>\n",
       "      <th>tube_weight_full</th>\n",
       "      <th>stool_weight_mg</th>\n",
       "      <th>cfu_g_stool</th>\n",
       "      <th>cfu_incubation_time</th>\n",
       "      <th>dna_conc_ng_ul</th>\n",
       "      <th>dna_conc_ng_ul.1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>s63</td>\n",
       "      <td>ILL134</td>\n",
       "      <td>s63-E1</td>\n",
       "      <td>P4C3T1</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>s63</td>\n",
       "      <td>ILL134</td>\n",
       "      <td>s63-E6</td>\n",
       "      <td>P4C3T1</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   sample_id     ill file_name project sample_type  date_collected  \\\n",
       "63       s63  ILL134    s63-E1  P4C3T1  Gavage DNA             NaN   \n",
       "64       s63  ILL134    s63-E6  P4C3T1  Gavage DNA             NaN   \n",
       "\n",
       "   time_point_h cage_id mouse_type  pretreatment  ... strain_bcs  \\\n",
       "63         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "64         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "\n",
       "           strains     tube_type tube_weight_empty tube_weight_full  \\\n",
       "63  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "64  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "\n",
       "   stool_weight_mg  cfu_g_stool  cfu_incubation_time  dna_conc_ng_ul  \\\n",
       "63             NaN          NaN                  NaN             NaN   \n",
       "64             NaN          NaN                  NaN             NaN   \n",
       "\n",
       "    dna_conc_ng_ul.1  \n",
       "63               NaN  \n",
       "64               NaN  \n",
       "\n",
       "[2 rows x 23 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10^4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>ill</th>\n",
       "      <th>file_name</th>\n",
       "      <th>project</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>date_collected</th>\n",
       "      <th>time_point_h</th>\n",
       "      <th>cage_id</th>\n",
       "      <th>mouse_type</th>\n",
       "      <th>pretreatment</th>\n",
       "      <th>...</th>\n",
       "      <th>strain_bcs</th>\n",
       "      <th>strains</th>\n",
       "      <th>tube_type</th>\n",
       "      <th>tube_weight_empty</th>\n",
       "      <th>tube_weight_full</th>\n",
       "      <th>stool_weight_mg</th>\n",
       "      <th>cfu_g_stool</th>\n",
       "      <th>cfu_incubation_time</th>\n",
       "      <th>dna_conc_ng_ul</th>\n",
       "      <th>dna_conc_ng_ul.1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [sample_id, ill, file_name, project, sample_type, date_collected, time_point_h, cage_id, mouse_type, pretreatment, mouse_id, gender, gavage_dose_200ul, strain_bcs, strains, tube_type, tube_weight_empty, tube_weight_full, stool_weight_mg, cfu_g_stool, cfu_incubation_time, dna_conc_ng_ul, dna_conc_ng_ul.1]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 23 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "ename": "KeyError",
     "evalue": "'strain'",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mKeyError\u001b[39m                                  Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[71]\u001b[39m\u001b[32m, line 39\u001b[39m\n\u001b[32m     37\u001b[39m \u001b[38;5;66;03m#pivot all dataframes\u001b[39;00m\n\u001b[32m     38\u001b[39m df_sample_pivot = df_sample_reads.pivot(index = [\u001b[33m'\u001b[39m\u001b[33mstrain\u001b[39m\u001b[33m'\u001b[39m, \u001b[33m'\u001b[39m\u001b[33mumi_seq\u001b[39m\u001b[33m'\u001b[39m], columns = \u001b[33m'\u001b[39m\u001b[33msample\u001b[39m\u001b[33m'\u001b[39m, values = \u001b[33m'\u001b[39m\u001b[33mmolecules\u001b[39m\u001b[33m'\u001b[39m).fillna(\u001b[32m0\u001b[39m)\n\u001b[32m---> \u001b[39m\u001b[32m39\u001b[39m df_gavage_pivot = \u001b[43mdf_reads_gavage\u001b[49m\u001b[43m.\u001b[49m\u001b[43mpivot\u001b[49m\u001b[43m(\u001b[49m\u001b[43mindex\u001b[49m\u001b[43m \u001b[49m\u001b[43m=\u001b[49m\u001b[43m \u001b[49m\u001b[43m[\u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43mstrain\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43mumi_seq\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcolumns\u001b[49m\u001b[43m \u001b[49m\u001b[43m=\u001b[49m\u001b[43m \u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43msample\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mvalues\u001b[49m\u001b[43m \u001b[49m\u001b[43m=\u001b[49m\u001b[43m \u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43mmolecules\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m)\u001b[49m.fillna(\u001b[32m0\u001b[39m)\n\u001b[32m     41\u001b[39m df_readstable = pd.concat([df_gavage_pivot, df_sample_pivot, df_ntc_pivot], axis = \u001b[32m1\u001b[39m).fillna(\u001b[32m0\u001b[39m)\n\u001b[32m     43\u001b[39m df_readstable.columns = df_readstable.columns.str.replace(\u001b[33m'\u001b[39m\u001b[33m \u001b[39m\u001b[33m'\u001b[39m, \u001b[33m'\u001b[39m\u001b[33m-\u001b[39m\u001b[33m'\u001b[39m)\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/opt/anaconda3/envs/pouch/lib/python3.13/site-packages/pandas/core/frame.py:9366\u001b[39m, in \u001b[36mDataFrame.pivot\u001b[39m\u001b[34m(self, columns, index, values)\u001b[39m\n\u001b[32m   9359\u001b[39m \u001b[38;5;129m@Substitution\u001b[39m(\u001b[33m\"\u001b[39m\u001b[33m\"\u001b[39m)\n\u001b[32m   9360\u001b[39m \u001b[38;5;129m@Appender\u001b[39m(_shared_docs[\u001b[33m\"\u001b[39m\u001b[33mpivot\u001b[39m\u001b[33m\"\u001b[39m])\n\u001b[32m   9361\u001b[39m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[34mpivot\u001b[39m(\n\u001b[32m   9362\u001b[39m     \u001b[38;5;28mself\u001b[39m, *, columns, index=lib.no_default, values=lib.no_default\n\u001b[32m   9363\u001b[39m ) -> DataFrame:\n\u001b[32m   9364\u001b[39m     \u001b[38;5;28;01mfrom\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[34;01mpandas\u001b[39;00m\u001b[34;01m.\u001b[39;00m\u001b[34;01mcore\u001b[39;00m\u001b[34;01m.\u001b[39;00m\u001b[34;01mreshape\u001b[39;00m\u001b[34;01m.\u001b[39;00m\u001b[34;01mpivot\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;28;01mimport\u001b[39;00m pivot\n\u001b[32m-> \u001b[39m\u001b[32m9366\u001b[39m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mpivot\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mindex\u001b[49m\u001b[43m=\u001b[49m\u001b[43mindex\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcolumns\u001b[49m\u001b[43m=\u001b[49m\u001b[43mcolumns\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mvalues\u001b[49m\u001b[43m=\u001b[49m\u001b[43mvalues\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/opt/anaconda3/envs/pouch/lib/python3.13/site-packages/pandas/core/reshape/pivot.py:553\u001b[39m, in \u001b[36mpivot\u001b[39m\u001b[34m(data, columns, index, values)\u001b[39m\n\u001b[32m    549\u001b[39m         index_list = [\n\u001b[32m    550\u001b[39m             data._constructor_sliced(data.index, name=data.index.name)\n\u001b[32m    551\u001b[39m         ]\n\u001b[32m    552\u001b[39m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[32m--> \u001b[39m\u001b[32m553\u001b[39m     index_list = [\u001b[43mdata\u001b[49m\u001b[43m[\u001b[49m\u001b[43midx\u001b[49m\u001b[43m]\u001b[49m \u001b[38;5;28;01mfor\u001b[39;00m idx \u001b[38;5;129;01min\u001b[39;00m com.convert_to_list_like(index)]\n\u001b[32m    555\u001b[39m data_columns = [data[col] \u001b[38;5;28;01mfor\u001b[39;00m col \u001b[38;5;129;01min\u001b[39;00m columns_listlike]\n\u001b[32m    556\u001b[39m index_list.extend(data_columns)\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/opt/anaconda3/envs/pouch/lib/python3.13/site-packages/pandas/core/frame.py:4113\u001b[39m, in \u001b[36mDataFrame.__getitem__\u001b[39m\u001b[34m(self, key)\u001b[39m\n\u001b[32m   4111\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m.columns.nlevels > \u001b[32m1\u001b[39m:\n\u001b[32m   4112\u001b[39m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m._getitem_multilevel(key)\n\u001b[32m-> \u001b[39m\u001b[32m4113\u001b[39m indexer = \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43mcolumns\u001b[49m\u001b[43m.\u001b[49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m   4114\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m is_integer(indexer):\n\u001b[32m   4115\u001b[39m     indexer = [indexer]\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/opt/anaconda3/envs/pouch/lib/python3.13/site-packages/pandas/core/indexes/range.py:417\u001b[39m, in \u001b[36mRangeIndex.get_loc\u001b[39m\u001b[34m(self, key)\u001b[39m\n\u001b[32m    415\u001b[39m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(key) \u001b[38;5;28;01mfrom\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[34;01merr\u001b[39;00m\n\u001b[32m    416\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(key, Hashable):\n\u001b[32m--> \u001b[39m\u001b[32m417\u001b[39m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(key)\n\u001b[32m    418\u001b[39m \u001b[38;5;28mself\u001b[39m._check_indexing_error(key)\n\u001b[32m    419\u001b[39m \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(key)\n",
      "\u001b[31mKeyError\u001b[39m: 'strain'"
     ]
    }
   ],
   "source": [
    "# generate readtables for getFP\n",
    "basepath = '/Volumes/sd/faith/MTCSB/projects/P4-barcoding_strains/r2_data/population-bottlenecks'\n",
    "exportpath = '/Volumes/sd/faith/MTCSB/projects/P4-barcoding_strains/r2_data/population-bottlenecks/fp_inputs/P4C3T1'\n",
    "li_strains = ['ST1', 'ST2']\n",
    "\n",
    "df_ntc = pd.DataFrame()\n",
    "for idx, row in df_ill_ntc.iterrows():\n",
    "    df = pd.read_csv(basepath + '/' + row['ill'] + '/readtables/' + row['file_name'] + \"/\" + row['file_name'] + '_total_moleculestable.csv')\n",
    "    df['sample'] = row['Sample Name/Pool Name*']+'_'+row['sample_id']\n",
    "    df_ntc = pd.concat([df_ntc, df])\n",
    "df_ntc_pivot = df_ntc.pivot(index = ['strain', 'umi_seq'], columns = 'sample', values = 'molecules').fillna(0)\n",
    "\n",
    "for dose in df_merge['gavage_dose_200ul'].unique():\n",
    "    print(dose)\n",
    "    df_dose = df_merge.loc[df_merge['gavage_dose_200ul']==dose]\n",
    "    #get gavage reads dataframe\n",
    "    #drop files without sequencing data\n",
    "    df_merge.dropna(subset=['file_name'], inplace = True)\n",
    "    df_gavage = df_dose[df_dose['sample_type'] == 'Gavage DNA'].copy()\n",
    "    display(df_gavage)\n",
    "    df_reads_gavage = pd.DataFrame()\n",
    "    for idx, row in df_gavage.iterrows():\n",
    "        df = pd.read_csv(basepath + '/' + row['ill'] + '/readtables/' + row['file_name'] + \"/\" + row['file_name'] + '_total_moleculestable.csv')\n",
    "        df['sample'] = row['project']+'_'+row['file_name']+'_'+row['sample_type']\n",
    "        df_reads_gavage = pd.concat([df_reads_gavage,df])\n",
    "\n",
    "    #get sample reads dataframe\n",
    "    #get all samples into dataframe   \n",
    "    df_samples = df_dose[df_dose['sample_type'] != 'Gavage DNA'].copy()\n",
    "    #print(df_samples)\n",
    "    df_sample_reads = pd.DataFrame()\n",
    "    for idx, row in df_samples.iterrows():\n",
    "        df = pd.read_csv(basepath + '/' + row['ill'] + '/readtables/' + row['file_name'] + \"/\" + row['file_name'] + '_total_moleculestable.csv')\n",
    "        df['sample'] = row['project']+'_'+row['file_name']+'_'+row['sample_type']\n",
    "        df_sample_reads = pd.concat([df_sample_reads,df])\n",
    "\n",
    "    #pivot all dataframes\n",
    "    df_sample_pivot = df_sample_reads.pivot(index = ['strain', 'umi_seq'], columns = 'sample', values = 'molecules').fillna(0)\n",
    "    df_gavage_pivot = df_reads_gavage.pivot(index = ['strain', 'umi_seq'], columns = 'sample', values = 'molecules').fillna(0)\n",
    "\n",
    "    df_readstable = pd.concat([df_gavage_pivot, df_sample_pivot, df_ntc_pivot], axis = 1).fillna(0)\n",
    "\n",
    "    df_readstable.columns = df_readstable.columns.str.replace(' ', '-')\n",
    "    df_readstable.reset_index(inplace = True)\n",
    "\n",
    "    for strain in li_strains:\n",
    "        df_readstable.loc[df_readstable['strain'] == strain].drop('strain', axis = 1).to_csv(exportpath + '/' + dose + '_' + strain + '_moleculestable.csv', index = False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "bdf03ffb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# read all readtables and drop rows in which all gavage columns are 0\n",
    "li_strains = ['ST1', 'ST2']\n",
    "exportpath = '/Volumes/sd/faith/MTCSB/projects/P4-barcoding_strains/r2_data/population-bottlenecks/fp_inputs/P4C3T1'\n",
    "for dose in ['10^6', '10^8']:\n",
    "    for strain in li_strains:\n",
    "        df = pd.read_csv(exportpath + '/'+ dose + '_'+strain + '_moleculestable.csv')\n",
    "        #get gavage columns\n",
    "        li_gavage = [col for col in df.columns if 'Gavage' in col]\n",
    "        #drop rows in which all gavage columns are 0\n",
    "        df[df[li_gavage].sum(axis = 1) > 0].to_csv(exportpath + '/'+ dose + '_'+  strain + '_onlygavage_moleculestable.csv', index = False)\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "248b6377",
   "metadata": {},
   "source": [
    "# P4C3T2 10k barcodes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "43deb6a1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>ill</th>\n",
       "      <th>file_name</th>\n",
       "      <th>project</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>date_collected</th>\n",
       "      <th>time_point_h</th>\n",
       "      <th>cage_id</th>\n",
       "      <th>mouse_type</th>\n",
       "      <th>pretreatment</th>\n",
       "      <th>...</th>\n",
       "      <th>strain_bcs</th>\n",
       "      <th>strains</th>\n",
       "      <th>tube_type</th>\n",
       "      <th>tube_weight_empty</th>\n",
       "      <th>tube_weight_full</th>\n",
       "      <th>stool_weight_mg</th>\n",
       "      <th>cfu_g_stool</th>\n",
       "      <th>cfu_incubation_time</th>\n",
       "      <th>dna_conc_ng_ul</th>\n",
       "      <th>dna_conc_ng_ul.1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>s1</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s1-A1</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5 h</td>\n",
       "      <td>cage1</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>1651.8</td>\n",
       "      <td>1693.0</td>\n",
       "      <td>41.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>s3</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s3-B1</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5 h</td>\n",
       "      <td>cage2</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>s5</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s5-C1</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5 h</td>\n",
       "      <td>cage3</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>s7</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s7-D1</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5 h</td>\n",
       "      <td>cage4</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>s9</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s9-E1</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5 h</td>\n",
       "      <td>cage5</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>s11</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s11-F1</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5 h</td>\n",
       "      <td>cage6</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>s13</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s13-G1</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage1</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>s15</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s15-H1</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage2</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>s17</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s17-A2</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage3</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>s19</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s19-B2</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage4</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>s21</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s21-C2</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage5</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>s23</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s23-D2</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>stool DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage6</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>s25</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s25-E2</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>SI DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage1</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>s27</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s27-F2</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>SI DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage2</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>s29</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s29-G2</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>SI DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage3</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>s31</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s31-H2</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>SI DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage4</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>s33</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s33-A3</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>SI DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage5</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>s35</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s35-B3</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>SI DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage6</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>s37</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s37-C3</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Cecum DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage1</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>s39</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s39-D3</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Cecum DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage2</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>s41</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s41-E3</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Cecum DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage3</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>s43</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s43-F3</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Cecum DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage4</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>s45</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s45-G3</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Cecum DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage5</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>s47</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s47-A4</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Cecum DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage6</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>s49</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s49-B4</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Pr. Colon DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage1</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>s51</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s51-C4</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Pr. Colon DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage2</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>s53</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s53-D4</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Pr. Colon DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage3</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>s55</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s55-E4</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Pr. Colon DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage4</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>s57</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s57-F4</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Pr. Colon DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage5</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>s59</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s59-G4</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Pr. Colon DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>cage6</td>\n",
       "      <td>germ_free</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>s61</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s61-H4</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>s62</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s62-A6</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>s63</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s63-B5</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>s64</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s64-C5</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>s65</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s65-D6</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>s66</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s66-E5</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>36 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   sample_id     ill file_name project    sample_type  date_collected  \\\n",
       "0         s1  ILL138     s1-A1  P4C3T2      stool DNA             NaN   \n",
       "1         s3  ILL138     s3-B1  P4C3T2      stool DNA             NaN   \n",
       "2         s5  ILL138     s5-C1  P4C3T2      stool DNA             NaN   \n",
       "3         s7  ILL138     s7-D1  P4C3T2      stool DNA             NaN   \n",
       "4         s9  ILL138     s9-E1  P4C3T2      stool DNA             NaN   \n",
       "5        s11  ILL138    s11-F1  P4C3T2      stool DNA             NaN   \n",
       "6        s13  ILL138    s13-G1  P4C3T2      stool DNA             NaN   \n",
       "7        s15  ILL138    s15-H1  P4C3T2      stool DNA             NaN   \n",
       "8        s17  ILL138    s17-A2  P4C3T2      stool DNA             NaN   \n",
       "9        s19  ILL138    s19-B2  P4C3T2      stool DNA             NaN   \n",
       "10       s21  ILL138    s21-C2  P4C3T2      stool DNA             NaN   \n",
       "11       s23  ILL138    s23-D2  P4C3T2      stool DNA             NaN   \n",
       "12       s25  ILL138    s25-E2  P4C3T2         SI DNA             NaN   \n",
       "13       s27  ILL138    s27-F2  P4C3T2         SI DNA             NaN   \n",
       "14       s29  ILL138    s29-G2  P4C3T2         SI DNA             NaN   \n",
       "15       s31  ILL138    s31-H2  P4C3T2         SI DNA             NaN   \n",
       "16       s33  ILL138    s33-A3  P4C3T2         SI DNA             NaN   \n",
       "17       s35  ILL138    s35-B3  P4C3T2         SI DNA             NaN   \n",
       "18       s37  ILL138    s37-C3  P4C3T2      Cecum DNA             NaN   \n",
       "19       s39  ILL138    s39-D3  P4C3T2      Cecum DNA             NaN   \n",
       "20       s41  ILL138    s41-E3  P4C3T2      Cecum DNA             NaN   \n",
       "21       s43  ILL138    s43-F3  P4C3T2      Cecum DNA             NaN   \n",
       "22       s45  ILL138    s45-G3  P4C3T2      Cecum DNA             NaN   \n",
       "23       s47  ILL138    s47-A4  P4C3T2      Cecum DNA             NaN   \n",
       "24       s49  ILL138    s49-B4  P4C3T2  Pr. Colon DNA             NaN   \n",
       "25       s51  ILL138    s51-C4  P4C3T2  Pr. Colon DNA             NaN   \n",
       "26       s53  ILL138    s53-D4  P4C3T2  Pr. Colon DNA             NaN   \n",
       "27       s55  ILL138    s55-E4  P4C3T2  Pr. Colon DNA             NaN   \n",
       "28       s57  ILL138    s57-F4  P4C3T2  Pr. Colon DNA             NaN   \n",
       "29       s59  ILL138    s59-G4  P4C3T2  Pr. Colon DNA             NaN   \n",
       "30       s61  ILL138    s61-H4  P4C3T2     Gavage DNA             NaN   \n",
       "31       s62  ILL138    s62-A6  P4C3T2     Gavage DNA             NaN   \n",
       "32       s63  ILL138    s63-B5  P4C3T2     Gavage DNA             NaN   \n",
       "33       s64  ILL138    s64-C5  P4C3T2     Gavage DNA             NaN   \n",
       "34       s65  ILL138    s65-D6  P4C3T2     Gavage DNA             NaN   \n",
       "35       s66  ILL138    s66-E5  P4C3T2     Gavage DNA             NaN   \n",
       "\n",
       "   time_point_h cage_id mouse_type  pretreatment  ... strain_bcs  \\\n",
       "0           5 h   cage1  germ_free           NaN  ...  ST1a|ST2d   \n",
       "1           5 h   cage2  germ_free           NaN  ...  ST1a|ST2d   \n",
       "2           5 h   cage3  germ_free           NaN  ...  ST1a|ST2d   \n",
       "3           5 h   cage4  germ_free           NaN  ...  ST1a|ST2d   \n",
       "4           5 h   cage5  germ_free           NaN  ...  ST1a|ST2d   \n",
       "5           5 h   cage6  germ_free           NaN  ...  ST1a|ST2d   \n",
       "6          24 h   cage1  germ_free           NaN  ...  ST1a|ST2d   \n",
       "7          24 h   cage2  germ_free           NaN  ...  ST1a|ST2d   \n",
       "8          24 h   cage3  germ_free           NaN  ...  ST1a|ST2d   \n",
       "9          24 h   cage4  germ_free           NaN  ...  ST1a|ST2d   \n",
       "10         24 h   cage5  germ_free           NaN  ...  ST1a|ST2d   \n",
       "11         24 h   cage6  germ_free           NaN  ...  ST1a|ST2d   \n",
       "12         24 h   cage1  germ_free           NaN  ...  ST1a|ST2d   \n",
       "13         24 h   cage2  germ_free           NaN  ...  ST1a|ST2d   \n",
       "14         24 h   cage3  germ_free           NaN  ...  ST1a|ST2d   \n",
       "15         24 h   cage4  germ_free           NaN  ...  ST1a|ST2d   \n",
       "16         24 h   cage5  germ_free           NaN  ...  ST1a|ST2d   \n",
       "17         24 h   cage6  germ_free           NaN  ...  ST1a|ST2d   \n",
       "18         24 h   cage1  germ_free           NaN  ...  ST1a|ST2d   \n",
       "19         24 h   cage2  germ_free           NaN  ...  ST1a|ST2d   \n",
       "20         24 h   cage3  germ_free           NaN  ...  ST1a|ST2d   \n",
       "21         24 h   cage4  germ_free           NaN  ...  ST1a|ST2d   \n",
       "22         24 h   cage5  germ_free           NaN  ...  ST1a|ST2d   \n",
       "23         24 h   cage6  germ_free           NaN  ...  ST1a|ST2d   \n",
       "24         24 h   cage1  germ_free           NaN  ...  ST1a|ST2d   \n",
       "25         24 h   cage2  germ_free           NaN  ...  ST1a|ST2d   \n",
       "26         24 h   cage3  germ_free           NaN  ...  ST1a|ST2d   \n",
       "27         24 h   cage4  germ_free           NaN  ...  ST1a|ST2d   \n",
       "28         24 h   cage5  germ_free           NaN  ...  ST1a|ST2d   \n",
       "29         24 h   cage6  germ_free           NaN  ...  ST1a|ST2d   \n",
       "30         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "31         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "32         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "33         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "34         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "35         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "\n",
       "           strains     tube_type tube_weight_empty tube_weight_full  \\\n",
       "0   E. coli|E.coli  2ml_screwcap            1651.8           1693.0   \n",
       "1   E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "2   E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "3   E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "4   E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "5   E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "6   E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "7   E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "8   E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "9   E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "10  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "11  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "12  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "13  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "14  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "15  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "16  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "17  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "18  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "19  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "20  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "21  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "22  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "23  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "24  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "25  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "26  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "27  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "28  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "29  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "30  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "31  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "32  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "33  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "34  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "35  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "\n",
       "   stool_weight_mg  cfu_g_stool  cfu_incubation_time  dna_conc_ng_ul  \\\n",
       "0             41.2          NaN                  NaN             NaN   \n",
       "1              NaN          NaN                  NaN             NaN   \n",
       "2              NaN          NaN                  NaN             NaN   \n",
       "3              NaN          NaN                  NaN             NaN   \n",
       "4              NaN          NaN                  NaN             NaN   \n",
       "5              NaN          NaN                  NaN             NaN   \n",
       "6              NaN          NaN                  NaN             NaN   \n",
       "7              NaN          NaN                  NaN             NaN   \n",
       "8              NaN          NaN                  NaN             NaN   \n",
       "9              NaN          NaN                  NaN             NaN   \n",
       "10             NaN          NaN                  NaN             NaN   \n",
       "11             NaN          NaN                  NaN             NaN   \n",
       "12             NaN          NaN                  NaN             NaN   \n",
       "13             NaN          NaN                  NaN             NaN   \n",
       "14             NaN          NaN                  NaN             NaN   \n",
       "15             NaN          NaN                  NaN             NaN   \n",
       "16             NaN          NaN                  NaN             NaN   \n",
       "17             NaN          NaN                  NaN             NaN   \n",
       "18             NaN          NaN                  NaN             NaN   \n",
       "19             NaN          NaN                  NaN             NaN   \n",
       "20             NaN          NaN                  NaN             NaN   \n",
       "21             NaN          NaN                  NaN             NaN   \n",
       "22             NaN          NaN                  NaN             NaN   \n",
       "23             NaN          NaN                  NaN             NaN   \n",
       "24             NaN          NaN                  NaN             NaN   \n",
       "25             NaN          NaN                  NaN             NaN   \n",
       "26             NaN          NaN                  NaN             NaN   \n",
       "27             NaN          NaN                  NaN             NaN   \n",
       "28             NaN          NaN                  NaN             NaN   \n",
       "29             NaN          NaN                  NaN             NaN   \n",
       "30             NaN          NaN                  NaN             NaN   \n",
       "31             NaN          NaN                  NaN             NaN   \n",
       "32             NaN          NaN                  NaN             NaN   \n",
       "33             NaN          NaN                  NaN             NaN   \n",
       "34             NaN          NaN                  NaN             NaN   \n",
       "35             NaN          NaN                  NaN             NaN   \n",
       "\n",
       "    dna_conc_ng_ul.1  \n",
       "0                NaN  \n",
       "1                NaN  \n",
       "2                NaN  \n",
       "3                NaN  \n",
       "4                NaN  \n",
       "5                NaN  \n",
       "6                NaN  \n",
       "7                NaN  \n",
       "8                NaN  \n",
       "9                NaN  \n",
       "10               NaN  \n",
       "11               NaN  \n",
       "12               NaN  \n",
       "13               NaN  \n",
       "14               NaN  \n",
       "15               NaN  \n",
       "16               NaN  \n",
       "17               NaN  \n",
       "18               NaN  \n",
       "19               NaN  \n",
       "20               NaN  \n",
       "21               NaN  \n",
       "22               NaN  \n",
       "23               NaN  \n",
       "24               NaN  \n",
       "25               NaN  \n",
       "26               NaN  \n",
       "27               NaN  \n",
       "28               NaN  \n",
       "29               NaN  \n",
       "30               NaN  \n",
       "31               NaN  \n",
       "32               NaN  \n",
       "33               NaN  \n",
       "34               NaN  \n",
       "35               NaN  \n",
       "\n",
       "[36 rows x 23 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Sample Name/Pool Name*</th>\n",
       "      <th>Library Name*</th>\n",
       "      <th>i7 Index Name*</th>\n",
       "      <th>i7 Index Sequence*</th>\n",
       "      <th>i5 Index Name*</th>\n",
       "      <th>i5 Index Sequence*</th>\n",
       "      <th>Custom Read 1*</th>\n",
       "      <th>Custom Read 2*</th>\n",
       "      <th>Custom Index 1*</th>\n",
       "      <th>Custom Index 2*</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>project</th>\n",
       "      <th>ill</th>\n",
       "      <th>file_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>24</td>\n",
       "      <td>NTC_H3</td>\n",
       "      <td>ILL138_P4C3T2_10k</td>\n",
       "      <td>N7F15</td>\n",
       "      <td>AACGTTAG</td>\n",
       "      <td>S5R08</td>\n",
       "      <td>AACTCTCT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NTC</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>NTC-H3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>48</td>\n",
       "      <td>WATER_D2</td>\n",
       "      <td>ILL139_P4C1T8</td>\n",
       "      <td>N7F26</td>\n",
       "      <td>CAGCGTTG</td>\n",
       "      <td>S5R60</td>\n",
       "      <td>AACCGCAA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WATER</td>\n",
       "      <td>P4C1T8</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>WATER-D2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>53</td>\n",
       "      <td>WATER_B3</td>\n",
       "      <td>ILL139_P4C1T8</td>\n",
       "      <td>N7F27</td>\n",
       "      <td>GCCATAGT</td>\n",
       "      <td>S5R58</td>\n",
       "      <td>TCCGGCGG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WATER</td>\n",
       "      <td>P4C1T8</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>WATER-B3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>75</td>\n",
       "      <td>WATER_C6</td>\n",
       "      <td>ILL139_P4C1T8</td>\n",
       "      <td>N7F30</td>\n",
       "      <td>ATAGAGAC</td>\n",
       "      <td>S5R59</td>\n",
       "      <td>CAATGCCT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WATER</td>\n",
       "      <td>P4C1T8</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>WATER-C6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>99</td>\n",
       "      <td>WATER_F9</td>\n",
       "      <td>ILL139_P4C1T8</td>\n",
       "      <td>N7F33</td>\n",
       "      <td>TAGTAATT</td>\n",
       "      <td>S5R62</td>\n",
       "      <td>GCATGGCG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WATER</td>\n",
       "      <td>P4C1T8</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>WATER-F9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Unnamed: 0 Sample Name/Pool Name*      Library Name* i7 Index Name*  \\\n",
       "23          24                 NTC_H3  ILL138_P4C3T2_10k          N7F15   \n",
       "47          48               WATER_D2      ILL139_P4C1T8          N7F26   \n",
       "52          53               WATER_B3      ILL139_P4C1T8          N7F27   \n",
       "74          75               WATER_C6      ILL139_P4C1T8          N7F30   \n",
       "98          99               WATER_F9      ILL139_P4C1T8          N7F33   \n",
       "\n",
       "   i7 Index Sequence* i5 Index Name* i5 Index Sequence*  Custom Read 1*  \\\n",
       "23           AACGTTAG          S5R08           AACTCTCT             NaN   \n",
       "47           CAGCGTTG          S5R60           AACCGCAA             NaN   \n",
       "52           GCCATAGT          S5R58           TCCGGCGG             NaN   \n",
       "74           ATAGAGAC          S5R59           CAATGCCT             NaN   \n",
       "98           TAGTAATT          S5R62           GCATGGCG             NaN   \n",
       "\n",
       "    Custom Read 2*  Custom Index 1*  Custom Index 2* sample_id project  \\\n",
       "23             NaN              NaN              NaN       NTC  P4C3T2   \n",
       "47             NaN              NaN              NaN     WATER  P4C1T8   \n",
       "52             NaN              NaN              NaN     WATER  P4C1T8   \n",
       "74             NaN              NaN              NaN     WATER  P4C1T8   \n",
       "98             NaN              NaN              NaN     WATER  P4C1T8   \n",
       "\n",
       "       ill file_name  \n",
       "23  ILL138    NTC-H3  \n",
       "47  ILL138  WATER-D2  \n",
       "52  ILL138  WATER-B3  \n",
       "74  ILL138  WATER-C6  \n",
       "98  ILL138  WATER-F9  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#P4C2T4:\n",
    "df_ill = pd.read_excel('/Users/ruprec01/Documents/Faith_lab/Git/population_bottlenecks/input_files/ill_sheets/ILL138_GENEWIZ.xls',\n",
    "                            sheet_name = 'Individual Library Information', header = 1)\n",
    "df_samplesheet = pd.read_csv('/Users/ruprec01/Documents/Faith_lab/Git/population_bottlenecks/input_files/metadata/P4C3T2_cr_mouse_sample.tsv', sep = '\\t')\n",
    "\n",
    "df_samplesheet = df_samplesheet.loc[df_samplesheet['sample_type'].str.contains('DNA')].copy()\n",
    "df_ill = df_ill.dropna(subset=['Sample Name/Pool Name*'])\n",
    "df_ill['sample_id']= df_ill['Sample Name/Pool Name*'].apply(lambda x: str(x).split('_')[0])\n",
    "df_ill['project'] = df_ill['Library Name*'].apply(lambda x: str(x).split('_')[1])\n",
    "#correct library name in ILL138 run\n",
    "df_ill['ill'] = df_ill['Library Name*'].apply(lambda x:  'ILL138' if str(x).split('_')[0] == 'ILL139' else str(x).split('_')[0])\n",
    "df_ill['file_name'] = df_ill['Sample Name/Pool Name*'].apply(lambda x: str(x).replace('_', '-'))\n",
    "df_merge = pd.merge(df_ill[['sample_id', 'ill', 'file_name', 'project']], df_samplesheet, how = 'right', on = ['sample_id', 'project'])\n",
    "df_merge.head(10)\n",
    "\n",
    "df_ill_ntc = df_ill.loc[df_ill['Sample Name/Pool Name*'].str.contains(\"WATER|NTC\")]\n",
    "df_merge = df_merge.dropna(subset= ['file_name'])\n",
    "display(df_merge)\n",
    "display(df_ill_ntc)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "c10b135d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10^8\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>ill</th>\n",
       "      <th>file_name</th>\n",
       "      <th>project</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>date_collected</th>\n",
       "      <th>time_point_h</th>\n",
       "      <th>cage_id</th>\n",
       "      <th>mouse_type</th>\n",
       "      <th>pretreatment</th>\n",
       "      <th>...</th>\n",
       "      <th>strain_bcs</th>\n",
       "      <th>strains</th>\n",
       "      <th>tube_type</th>\n",
       "      <th>tube_weight_empty</th>\n",
       "      <th>tube_weight_full</th>\n",
       "      <th>stool_weight_mg</th>\n",
       "      <th>cfu_g_stool</th>\n",
       "      <th>cfu_incubation_time</th>\n",
       "      <th>dna_conc_ng_ul</th>\n",
       "      <th>dna_conc_ng_ul.1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>s61</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s61-H4</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>s62</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s62-A6</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   sample_id     ill file_name project sample_type  date_collected  \\\n",
       "30       s61  ILL138    s61-H4  P4C3T2  Gavage DNA             NaN   \n",
       "31       s62  ILL138    s62-A6  P4C3T2  Gavage DNA             NaN   \n",
       "\n",
       "   time_point_h cage_id mouse_type  pretreatment  ... strain_bcs  \\\n",
       "30         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "31         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "\n",
       "           strains     tube_type tube_weight_empty tube_weight_full  \\\n",
       "30  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "31  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "\n",
       "   stool_weight_mg  cfu_g_stool  cfu_incubation_time  dna_conc_ng_ul  \\\n",
       "30             NaN          NaN                  NaN             NaN   \n",
       "31             NaN          NaN                  NaN             NaN   \n",
       "\n",
       "    dna_conc_ng_ul.1  \n",
       "30               NaN  \n",
       "31               NaN  \n",
       "\n",
       "[2 rows x 23 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10^6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>ill</th>\n",
       "      <th>file_name</th>\n",
       "      <th>project</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>date_collected</th>\n",
       "      <th>time_point_h</th>\n",
       "      <th>cage_id</th>\n",
       "      <th>mouse_type</th>\n",
       "      <th>pretreatment</th>\n",
       "      <th>...</th>\n",
       "      <th>strain_bcs</th>\n",
       "      <th>strains</th>\n",
       "      <th>tube_type</th>\n",
       "      <th>tube_weight_empty</th>\n",
       "      <th>tube_weight_full</th>\n",
       "      <th>stool_weight_mg</th>\n",
       "      <th>cfu_g_stool</th>\n",
       "      <th>cfu_incubation_time</th>\n",
       "      <th>dna_conc_ng_ul</th>\n",
       "      <th>dna_conc_ng_ul.1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>s63</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s63-B5</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>s64</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s64-C5</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   sample_id     ill file_name project sample_type  date_collected  \\\n",
       "32       s63  ILL138    s63-B5  P4C3T2  Gavage DNA             NaN   \n",
       "33       s64  ILL138    s64-C5  P4C3T2  Gavage DNA             NaN   \n",
       "\n",
       "   time_point_h cage_id mouse_type  pretreatment  ... strain_bcs  \\\n",
       "32         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "33         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "\n",
       "           strains     tube_type tube_weight_empty tube_weight_full  \\\n",
       "32  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "33  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "\n",
       "   stool_weight_mg  cfu_g_stool  cfu_incubation_time  dna_conc_ng_ul  \\\n",
       "32             NaN          NaN                  NaN             NaN   \n",
       "33             NaN          NaN                  NaN             NaN   \n",
       "\n",
       "    dna_conc_ng_ul.1  \n",
       "32               NaN  \n",
       "33               NaN  \n",
       "\n",
       "[2 rows x 23 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10^4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>ill</th>\n",
       "      <th>file_name</th>\n",
       "      <th>project</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>date_collected</th>\n",
       "      <th>time_point_h</th>\n",
       "      <th>cage_id</th>\n",
       "      <th>mouse_type</th>\n",
       "      <th>pretreatment</th>\n",
       "      <th>...</th>\n",
       "      <th>strain_bcs</th>\n",
       "      <th>strains</th>\n",
       "      <th>tube_type</th>\n",
       "      <th>tube_weight_empty</th>\n",
       "      <th>tube_weight_full</th>\n",
       "      <th>stool_weight_mg</th>\n",
       "      <th>cfu_g_stool</th>\n",
       "      <th>cfu_incubation_time</th>\n",
       "      <th>dna_conc_ng_ul</th>\n",
       "      <th>dna_conc_ng_ul.1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>s65</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s65-D6</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>s66</td>\n",
       "      <td>ILL138</td>\n",
       "      <td>s66-E5</td>\n",
       "      <td>P4C3T2</td>\n",
       "      <td>Gavage DNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24 h</td>\n",
       "      <td>gavage</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ST1a|ST2d</td>\n",
       "      <td>E. coli|E.coli</td>\n",
       "      <td>2ml_screwcap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   sample_id     ill file_name project sample_type  date_collected  \\\n",
       "34       s65  ILL138    s65-D6  P4C3T2  Gavage DNA             NaN   \n",
       "35       s66  ILL138    s66-E5  P4C3T2  Gavage DNA             NaN   \n",
       "\n",
       "   time_point_h cage_id mouse_type  pretreatment  ... strain_bcs  \\\n",
       "34         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "35         24 h  gavage        NaN           NaN  ...  ST1a|ST2d   \n",
       "\n",
       "           strains     tube_type tube_weight_empty tube_weight_full  \\\n",
       "34  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "35  E. coli|E.coli  2ml_screwcap               NaN              NaN   \n",
       "\n",
       "   stool_weight_mg  cfu_g_stool  cfu_incubation_time  dna_conc_ng_ul  \\\n",
       "34             NaN          NaN                  NaN             NaN   \n",
       "35             NaN          NaN                  NaN             NaN   \n",
       "\n",
       "    dna_conc_ng_ul.1  \n",
       "34               NaN  \n",
       "35               NaN  \n",
       "\n",
       "[2 rows x 23 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# generate readtables for getFP\n",
    "basepath = '/Volumes/sd/faith/MTCSB/projects/P4-barcoding_strains/r2_data/population-bottlenecks'\n",
    "exportpath = '/Volumes/sd/faith/MTCSB/projects/P4-barcoding_strains/r2_data/population-bottlenecks/fp_inputs/P4C3T2'\n",
    "li_strains = ['ST1', 'ST2']\n",
    "\n",
    "df_ntc = pd.DataFrame()\n",
    "for idx, row in df_ill_ntc.iterrows():\n",
    "    df = pd.read_csv(basepath + '/' + row['ill'] + '/readtables/' + row['file_name'] + \"/\" + row['file_name'] + '_total_moleculestable.csv')\n",
    "    df['sample'] = row['Sample Name/Pool Name*']+'_'+row['sample_id']\n",
    "    df_ntc = pd.concat([df_ntc, df])\n",
    "df_ntc_pivot = df_ntc.pivot(index = ['strain', 'umi_seq'], columns = 'sample', values = 'molecules').fillna(0)\n",
    "\n",
    "for dose in df_merge['gavage_dose_200ul'].unique():\n",
    "    print(dose)\n",
    "    df_dose = df_merge.loc[df_merge['gavage_dose_200ul']==dose]\n",
    "    #get gavage reads dataframe\n",
    "    #drop files without sequencing data\n",
    "    df_merge.dropna(subset=['file_name'], inplace = True)\n",
    "    df_gavage = df_dose[df_dose['sample_type'] == 'Gavage DNA'].copy()\n",
    "    display(df_gavage)\n",
    "    df_reads_gavage = pd.DataFrame()\n",
    "    for idx, row in df_gavage.iterrows():\n",
    "        df = pd.read_csv(basepath + '/' + row['ill'] + '/readtables/' + row['file_name'] + \"/\" + row['file_name'] + '_total_moleculestable.csv')\n",
    "        df['sample'] = row['project']+'_'+row['file_name']+'_'+row['sample_type']\n",
    "        df_reads_gavage = pd.concat([df_reads_gavage,df])\n",
    "\n",
    "    #get sample reads dataframe\n",
    "    #get all samples into dataframe   \n",
    "    df_samples = df_dose[df_dose['sample_type'] != 'Gavage DNA'].copy()\n",
    "    #print(df_samples)\n",
    "    df_sample_reads = pd.DataFrame()\n",
    "    for idx, row in df_samples.iterrows():\n",
    "        df = pd.read_csv(basepath + '/' + row['ill'] + '/readtables/' + row['file_name'] + \"/\" + row['file_name'] + '_total_moleculestable.csv')\n",
    "        df['sample'] = row['project']+'_'+row['file_name']+'_'+row['sample_type']\n",
    "        df_sample_reads = pd.concat([df_sample_reads,df])\n",
    "\n",
    "    #pivot all dataframes\n",
    "    df_sample_pivot = df_sample_reads.pivot(index = ['strain', 'umi_seq'], columns = 'sample', values = 'molecules').fillna(0)\n",
    "    df_gavage_pivot = df_reads_gavage.pivot(index = ['strain', 'umi_seq'], columns = 'sample', values = 'molecules').fillna(0)\n",
    "\n",
    "    df_readstable = pd.concat([df_gavage_pivot, df_sample_pivot, df_ntc_pivot], axis = 1).fillna(0)\n",
    "\n",
    "    df_readstable.columns = df_readstable.columns.str.replace(' ', '-')\n",
    "    df_readstable.reset_index(inplace = True)\n",
    "\n",
    "    for strain in li_strains:\n",
    "        df_readstable.loc[df_readstable['strain'] == strain].drop('strain', axis = 1).to_csv(exportpath + '/' + dose + '_' + strain + '_moleculestable.csv', index = False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "b15a0fb1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# read all readtables and drop rows in which all gavage columns are 0\n",
    "li_strains = ['ST1', 'ST2']\n",
    "exportpath = '/Volumes/sd/faith/MTCSB/projects/P4-barcoding_strains/r2_data/population-bottlenecks/fp_inputs/P4C3T2'\n",
    "for dose in ['10^4','10^6', '10^8']:\n",
    "    for strain in li_strains:\n",
    "        df = pd.read_csv(exportpath + '/'+ dose + '_'+strain + '_moleculestable.csv')\n",
    "        #get gavage columns\n",
    "        li_gavage = [col for col in df.columns if 'Gavage' in col]\n",
    "        #drop rows in which all gavage columns are 0\n",
    "        df[df[li_gavage].sum(axis = 1) > 0].to_csv(exportpath + '/'+ dose + '_'+  strain + '_onlygavage_moleculestable.csv', index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "51411809",
   "metadata": {},
   "source": [
    "# P4C4T1 In vitro\n",
    " - takes gavage samples from P4C1T4T5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "d8cb348b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>ill</th>\n",
       "      <th>file_name</th>\n",
       "      <th>project</th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Row</th>\n",
       "      <th>Column</th>\n",
       "      <th>OD620</th>\n",
       "      <th>Sample</th>\n",
       "      <th>Inoculation</th>\n",
       "      <th>Inoculation_2</th>\n",
       "      <th>Inoculation_2_time</th>\n",
       "      <th>CmR</th>\n",
       "      <th>Media</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1</td>\n",
       "      <td>ILL132</td>\n",
       "      <td>A1</td>\n",
       "      <td>P4C4T1</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>0.384</td>\n",
       "      <td>Sample 2</td>\n",
       "      <td>barcodes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GAM+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2</td>\n",
       "      <td>ILL132</td>\n",
       "      <td>A2</td>\n",
       "      <td>P4C4T1</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0.363</td>\n",
       "      <td>Sample 10</td>\n",
       "      <td>barcodes</td>\n",
       "      <td>mtc01</td>\n",
       "      <td>3h</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GAM+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A3</td>\n",
       "      <td>ILL132</td>\n",
       "      <td>A3</td>\n",
       "      <td>P4C4T1</td>\n",
       "      <td>2</td>\n",
       "      <td>A</td>\n",
       "      <td>3</td>\n",
       "      <td>0.419</td>\n",
       "      <td>Sample 18</td>\n",
       "      <td>barcodes</td>\n",
       "      <td>mtc01</td>\n",
       "      <td>6h</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GAM+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A4</td>\n",
       "      <td>ILL132</td>\n",
       "      <td>A4</td>\n",
       "      <td>P4C4T1</td>\n",
       "      <td>3</td>\n",
       "      <td>A</td>\n",
       "      <td>4</td>\n",
       "      <td>0.426</td>\n",
       "      <td>Sample 26</td>\n",
       "      <td>mtc01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GAM+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A5</td>\n",
       "      <td>ILL132</td>\n",
       "      <td>A5</td>\n",
       "      <td>P4C4T1</td>\n",
       "      <td>4</td>\n",
       "      <td>A</td>\n",
       "      <td>5</td>\n",
       "      <td>0.424</td>\n",
       "      <td>Sample 34</td>\n",
       "      <td>mtc01</td>\n",
       "      <td>barcodes</td>\n",
       "      <td>3h</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GAM+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>H8</td>\n",
       "      <td>ILL132</td>\n",
       "      <td>H8</td>\n",
       "      <td>P4C4T1</td>\n",
       "      <td>91</td>\n",
       "      <td>H</td>\n",
       "      <td>8</td>\n",
       "      <td>0.128</td>\n",
       "      <td>NaN</td>\n",
       "      <td>barcodes</td>\n",
       "      <td>mtc01</td>\n",
       "      <td>3h</td>\n",
       "      <td>CmR 20</td>\n",
       "      <td>GAM-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>H9</td>\n",
       "      <td>ILL132</td>\n",
       "      <td>H9</td>\n",
       "      <td>P4C4T1</td>\n",
       "      <td>92</td>\n",
       "      <td>H</td>\n",
       "      <td>9</td>\n",
       "      <td>0.134</td>\n",
       "      <td>NaN</td>\n",
       "      <td>barcodes</td>\n",
       "      <td>mtc01</td>\n",
       "      <td>6h</td>\n",
       "      <td>CmR 20</td>\n",
       "      <td>GAM-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>H10</td>\n",
       "      <td>ILL132</td>\n",
       "      <td>H10</td>\n",
       "      <td>P4C4T1</td>\n",
       "      <td>93</td>\n",
       "      <td>H</td>\n",
       "      <td>10</td>\n",
       "      <td>0.047</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mtc01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CmR 20</td>\n",
       "      <td>GAM-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>H11</td>\n",
       "      <td>ILL132</td>\n",
       "      <td>H11</td>\n",
       "      <td>P4C4T1</td>\n",
       "      <td>94</td>\n",
       "      <td>H</td>\n",
       "      <td>11</td>\n",
       "      <td>0.069</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mtc01</td>\n",
       "      <td>barcodes</td>\n",
       "      <td>3h</td>\n",
       "      <td>CmR 20</td>\n",
       "      <td>GAM-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>H12</td>\n",
       "      <td>ILL132</td>\n",
       "      <td>H12</td>\n",
       "      <td>P4C4T1</td>\n",
       "      <td>95</td>\n",
       "      <td>H</td>\n",
       "      <td>12</td>\n",
       "      <td>0.047</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mtc01</td>\n",
       "      <td>barcodes</td>\n",
       "      <td>6h</td>\n",
       "      <td>CmR 20</td>\n",
       "      <td>GAM-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>96 rows Ã— 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   sample_id     ill file_name project  Unnamed: 0 Row  Column  OD620  \\\n",
       "0         A1  ILL132        A1  P4C4T1           0   A       1  0.384   \n",
       "1         A2  ILL132        A2  P4C4T1           1   A       2  0.363   \n",
       "2         A3  ILL132        A3  P4C4T1           2   A       3  0.419   \n",
       "3         A4  ILL132        A4  P4C4T1           3   A       4  0.426   \n",
       "4         A5  ILL132        A5  P4C4T1           4   A       5  0.424   \n",
       "..       ...     ...       ...     ...         ...  ..     ...    ...   \n",
       "91        H8  ILL132        H8  P4C4T1          91   H       8  0.128   \n",
       "92        H9  ILL132        H9  P4C4T1          92   H       9  0.134   \n",
       "93       H10  ILL132       H10  P4C4T1          93   H      10  0.047   \n",
       "94       H11  ILL132       H11  P4C4T1          94   H      11  0.069   \n",
       "95       H12  ILL132       H12  P4C4T1          95   H      12  0.047   \n",
       "\n",
       "       Sample Inoculation Inoculation_2 Inoculation_2_time     CmR Media  \n",
       "0    Sample 2    barcodes           NaN                NaN     NaN  GAM+  \n",
       "1   Sample 10    barcodes         mtc01                 3h     NaN  GAM+  \n",
       "2   Sample 18    barcodes         mtc01                 6h     NaN  GAM+  \n",
       "3   Sample 26       mtc01           NaN                NaN     NaN  GAM+  \n",
       "4   Sample 34       mtc01      barcodes                 3h     NaN  GAM+  \n",
       "..        ...         ...           ...                ...     ...   ...  \n",
       "91        NaN    barcodes         mtc01                 3h  CmR 20  GAM-  \n",
       "92        NaN    barcodes         mtc01                 6h  CmR 20  GAM-  \n",
       "93        NaN       mtc01           NaN                NaN  CmR 20  GAM-  \n",
       "94        NaN       mtc01      barcodes                 3h  CmR 20  GAM-  \n",
       "95        NaN       mtc01      barcodes                 6h  CmR 20  GAM-  \n",
       "\n",
       "[96 rows x 14 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Sample Name/Pool Name*</th>\n",
       "      <th>Library Name*</th>\n",
       "      <th>i7 Index Name*</th>\n",
       "      <th>i7 Index Sequence*</th>\n",
       "      <th>i7 RC</th>\n",
       "      <th>i5 Index Name*</th>\n",
       "      <th>i5 Index Sequence*</th>\n",
       "      <th>i5 RC</th>\n",
       "      <th>Custom Primer Information Information</th>\n",
       "      <th>Custom Read 2*</th>\n",
       "      <th>Custom Index 1*</th>\n",
       "      <th>Custom Index 2*</th>\n",
       "      <th>Unnamed: 13</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>project</th>\n",
       "      <th>ill</th>\n",
       "      <th>file_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Unnamed: 0, Sample Name/Pool Name*, Library Name*, i7 Index Name*, i7 Index Sequence*, i7 RC, i5 Index Name*, i5 Index Sequence*, i5 RC, Custom Primer Information Information, Custom Read 2*, Custom Index 1*, Custom Index 2*, Unnamed: 13, sample_id, project, ill, file_name]\n",
       "Index: []"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#P4C2T4:\n",
    "df_ill = pd.read_excel('/Users/ruprec01/Documents/Faith_lab/Git/population_bottlenecks/input_files/ill_sheets/ILL132_P4C4T1.xlsx',\n",
    "                            sheet_name = 'Individual Library Information', header = 1)\n",
    "\n",
    "df_samplesheet = pd.read_csv('/Users/ruprec01/Documents/Faith_lab/Git/population_bottlenecks/input_files/metadata/P4C4T1_metadata.tsv', sep = '\\t')\n",
    "df_samplesheet['project'] = 'P4C4T1'\n",
    "df_ill = df_ill.dropna(subset=['Sample Name/Pool Name*'])\n",
    "df_ill['sample_id']= df_ill['Sample Name/Pool Name*'].apply(lambda x: str(x).split('_')[0])\n",
    "df_ill['project'] = df_ill['Library Name*'].apply(lambda x: str(x).split('_')[1])\n",
    "#correct library name in ILL138 run\n",
    "df_ill['ill'] = df_ill['Library Name*'].apply(lambda x:  'ILL138' if str(x).split('_')[0] == 'ILL139' else str(x).split('_')[0])\n",
    "df_ill['file_name'] = df_ill['Sample Name/Pool Name*'].apply(lambda x: str(x).replace('_', '-'))\n",
    "df_merge = pd.merge(df_ill[['sample_id', 'ill', 'file_name', 'project']], df_samplesheet, how = 'right', on = ['sample_id', 'project'])\n",
    "df_merge.head(10)\n",
    "\n",
    "df_ill_ntc = df_ill.loc[df_ill['Sample Name/Pool Name*'].str.contains(\"WATER|NTC\")]\n",
    "df_merge = df_merge.dropna(subset= ['file_name'])\n",
    "display(df_merge)\n",
    "display(df_ill_ntc)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "de7c66d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# generate readtables for getFP\n",
    "basepath = '/Volumes/sd/faith/MTCSB/projects/P4-barcoding_strains/r2_data/population-bottlenecks'\n",
    "exportpath = '/Volumes/sd/faith/MTCSB/projects/P4-barcoding_strains/r2_data/population-bottlenecks/fp_inputs/P4C4T1'\n",
    "li_strains = ['ST1', 'ST2', 'ST4', 'ST5', 'ST6']\n",
    "\n",
    "\n",
    "#get gavage reads dataframe\n",
    "#drop files without sequencing data\n",
    "df_merge.dropna(subset=['file_name'], inplace = True)\n",
    "\n",
    "#print(df_samples)\n",
    "df_sample_reads = pd.DataFrame()\n",
    "for idx, row in df_merge.iterrows():\n",
    "    df = pd.read_csv(basepath + '/' + row['ill'] + '/readtables/' + row['file_name'] + \"/\" + row['file_name'] + '_total_moleculestable.csv')\n",
    "    df['sample'] = row['project']+'_'+row['file_name']\n",
    "    df_sample_reads = pd.concat([df_sample_reads,df])\n",
    "\n",
    "#pivot all dataframes\n",
    "df_sample_pivot = df_sample_reads.pivot(index = ['strain', 'umi_seq'], columns = 'sample', values = 'molecules').fillna(0)\n",
    "df_readstable = df_sample_pivot.copy()\n",
    "df_readstable.columns = df_readstable.columns.str.replace(' ', '-')\n",
    "df_readstable.reset_index(inplace = True)\n",
    "for strain in li_strains:\n",
    "    df_readstable.loc[df_readstable['strain'] == strain].drop('strain', axis = 1).to_csv(exportpath + '/' + strain + '_moleculestable.csv', index = False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "a8d21eb2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# read all readtables and drop rows in which all gavage columns are 0\n",
    "li_strains = ['ST1', 'ST2', 'ST4', 'ST5', 'ST6']\n",
    "gavagepath = '/Volumes/sd/faith/MTCSB/projects/P4-barcoding_strains/r2_data/population-bottlenecks/fp_inputs/P4C2T4T5'\n",
    "exportpath = '/Volumes/sd/faith/MTCSB/projects/P4-barcoding_strains/r2_data/population-bottlenecks/fp_inputs/P4C4T1'\n",
    "for strain in li_strains:\n",
    "    df_gavage = pd.read_csv(gavagepath + '/' + strain + '_onlygavage_moleculestable.csv')\n",
    "    #get gavage columns\n",
    "    li_gavage = [col for col in df.columns if 'Gavage' in col]\n",
    "    df = pd.read_csv(exportpath + '/' + strain + '_moleculestable.csv')\n",
    "    df.set_index('umi_seq', inplace = True)\n",
    "    df_gavage.set_index('umi_seq', inplace = True)\n",
    "    df = pd.concat([df_gavage[li_gavage] ,df], axis = 1)\n",
    "    df.reset_index(inplace = True)\n",
    "    df.fillna(0, inplace = True)\n",
    "    #drop rows in which all gavage columns are 0\n",
    "    df[df[li_gavage].sum(axis = 1) > 0].to_csv(exportpath + '/' + strain + '_onlygavage_moleculestable.csv', index = False)\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "57ab177d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pouch",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
